diff --git a/src/types/BeforeSwapDelta.sol b/src/types/BeforeSwapDelta.sol
index 48ff5d82..f2fc7b1a 100644
--- a/src/types/BeforeSwapDelta.sol
+++ b/src/types/BeforeSwapDelta.sol
@@ -1,5 +1,7 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.20;
+pragma solidity ^0.8.0;
+
+import {BalanceDelta} from "./BalanceDelta.sol";
 
 // Return type of the beforeSwap hook.
 // Upper 128 bits is the delta in specified tokens. Lower 128 bits is delta in unspecified tokens (to match the afterSwap hook)
@@ -21,7 +23,7 @@ library BeforeSwapDeltaLibrary {
     /// extracts int128 from the upper 128 bits of the BeforeSwapDelta
     /// returned by beforeSwap
     function getSpecifiedDelta(BeforeSwapDelta delta) internal pure returns (int128 deltaSpecified) {
-        assembly ("memory-safe") {
+        assembly {
             deltaSpecified := sar(128, delta)
         }
     }
