diff --git a/test/foundry-tests/BitMath.t.sol b/test/foundry-tests/BitMath.t.sol
index f4be8eb5..41694cc7 100644
--- a/test/foundry-tests/BitMath.t.sol
+++ b/test/foundry-tests/BitMath.t.sol
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.8.19;
+pragma solidity ^0.8.15;
 
 import {GasSnapshot} from "forge-gas-snapshot/GasSnapshot.sol";
 import {Test} from "forge-std/Test.sol";
