diff --git a/contracts/libraries/BitMath.sol b/contracts/libraries/BitMath.sol
index 2575fe92..a26bb59e 100644
--- a/contracts/libraries/BitMath.sol
+++ b/contracts/libraries/BitMath.sol
@@ -54,27 +54,27 @@ library BitMath {
         require(x > 0, 'LSB');
 
         r = 255;
-        if (x & uint128(-1) > 0) {
+        if (x & type(uint128).max > 0) {
             r -= 128;
         } else {
             x >>= 128;
         }
-        if (x & uint64(-1) > 0) {
+        if (x & type(uint64).max > 0) {
             r -= 64;
         } else {
             x >>= 64;
         }
-        if (x & uint32(-1) > 0) {
+        if (x & type(uint32).max > 0) {
             r -= 32;
         } else {
             x >>= 32;
         }
-        if (x & uint16(-1) > 0) {
+        if (x & type(uint16).max > 0) {
             r -= 16;
         } else {
             x >>= 16;
         }
-        if (x & uint8(-1) > 0) {
+        if (x & type(uint8).max > 0) {
             r -= 8;
         } else {
             x >>= 8;
