diff --git a/contracts/test/BitMathEchidnaTest.sol b/contracts/test/BitMathEchidnaTest.sol
index 4519f530..09583c01 100644
--- a/contracts/test/BitMathEchidnaTest.sol
+++ b/contracts/test/BitMathEchidnaTest.sol
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity =0.8.19;
 
-import {BitMath} from "../libraries/BitMath.sol";
+import {BitMath} from '../libraries/BitMath.sol';
 
 contract BitMathEchidnaTest {
     function mostSignificantBitInvariant(uint256 input) external pure {
