diff --git a/test/utils/Constants.sol b/test/utils/Constants.sol
index b60b1cec..dbe6b6c9 100644
--- a/test/utils/Constants.sol
+++ b/test/utils/Constants.sol
@@ -14,9 +14,8 @@ library Constants {
     uint160 constant MAX_UINT160 = type(uint160).max;
 
     address constant ADDRESS_ZERO = address(0);
-
-    /// 0011 1111 1111 1111
-    address payable constant ALL_HOOKS = payable(0x0000000000000000000000000000000000003fFF);
+    address constant EMPTY_HOOKS = address(0x0000000000000000000000000000000000003c00);
+    address constant MOCK_HOOKS = address(0x0000000000000000000000000000000000003Fc0);
 
     uint256 constant POOL_SLOT = 10;
     uint256 constant TICKS_OFFSET = 4;
