diff --git a/test/utils/CurrencySettler.sol b/test/utils/CurrencySettler.sol
index 71d3d3d3..18c5684a 100644
--- a/test/utils/CurrencySettler.sol
+++ b/test/utils/CurrencySettler.sol
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.20;
 
-import {Currency} from "../../src/types/Currency.sol";
+import {Currency, CurrencyLibrary} from "../../src/types/Currency.sol";
 import {IERC20Minimal} from "../../src/interfaces/external/IERC20Minimal.sol";
 import {IPoolManager} from "../../src/interfaces/IPoolManager.sol";
 
@@ -10,6 +10,8 @@ import {IPoolManager} from "../../src/interfaces/IPoolManager.sol";
 /// To settle a negative delta (a debt on the user), a user make transfer or burn to pay off a debt.
 /// @dev Note that sync() is called before any erc-20 transfer in `settle`.
 library CurrencySettler {
+    using CurrencyLibrary for Currency;
+
     /// @notice Settle (pay) a currency to the PoolManager
     /// @param currency Currency to settle
     /// @param manager IPoolManager to settle to
