diff --git a/src/types/Currency.sol b/src/types/Currency.sol
index f8c5665c..59326a7f 100644
--- a/src/types/Currency.sol
+++ b/src/types/Currency.sol
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.20;
+pragma solidity ^0.8.0;
 
 import {IERC20Minimal} from "../interfaces/external/IERC20Minimal.sol";
 
@@ -89,10 +89,6 @@ library CurrencyLibrary {
         return Currency.unwrap(currency) == Currency.unwrap(NATIVE);
     }
 
-    function isZero(Currency currency) internal pure returns (bool) {
-        return isNative(currency);
-    }
-
     function toId(Currency currency) internal pure returns (uint256) {
         return uint160(Currency.unwrap(currency));
     }
