diff --git a/src/test/CustomCurveHook.sol b/src/test/CustomCurveHook.sol
index 2ca28875..e1fbf227 100644
--- a/src/test/CustomCurveHook.sol
+++ b/src/test/CustomCurveHook.sol
@@ -12,10 +12,11 @@ import {Currency} from "../types/Currency.sol";
 import {CurrencySettler} from "../../test/utils/CurrencySettler.sol";
 import {BaseTestHooks} from "./BaseTestHooks.sol";
 import {IERC20Minimal} from "../interfaces/external/IERC20Minimal.sol";
-import {Currency} from "../types/Currency.sol";
+import {CurrencyLibrary, Currency} from "../types/Currency.sol";
 
 contract CustomCurveHook is BaseTestHooks {
     using Hooks for IHooks;
+    using CurrencyLibrary for Currency;
     using CurrencySettler for Currency;
 
     error AddLiquidityDirectToHook();
