diff --git a/src/libraries/CustomRevert.sol b/src/libraries/CustomRevert.sol
index 3736b095..37483e8f 100644
--- a/src/libraries/CustomRevert.sol
+++ b/src/libraries/CustomRevert.sol
@@ -19,7 +19,7 @@ library CustomRevert {
     function revertWith(bytes4 selector, address addr) internal pure {
         assembly ("memory-safe") {
             mstore(0, selector)
-            mstore(0x04, and(addr, 0xffffffffffffffffffffffffffffffffffffffff))
+            mstore(0x04, addr)
             revert(0, 0x24)
         }
     }
@@ -28,7 +28,7 @@ library CustomRevert {
     function revertWith(bytes4 selector, int24 value) internal pure {
         assembly ("memory-safe") {
             mstore(0, selector)
-            mstore(0x04, signextend(2, value))
+            mstore(0x04, value)
             revert(0, 0x24)
         }
     }
@@ -37,7 +37,7 @@ library CustomRevert {
     function revertWith(bytes4 selector, uint160 value) internal pure {
         assembly ("memory-safe") {
             mstore(0, selector)
-            mstore(0x04, and(value, 0xffffffffffffffffffffffffffffffffffffffff))
+            mstore(0x04, value)
             revert(0, 0x24)
         }
     }
@@ -47,8 +47,8 @@ library CustomRevert {
         assembly ("memory-safe") {
             let fmp := mload(0x40)
             mstore(fmp, selector)
-            mstore(add(fmp, 0x04), signextend(2, value1))
-            mstore(add(fmp, 0x24), signextend(2, value2))
+            mstore(add(fmp, 0x04), value1)
+            mstore(add(fmp, 0x24), value2)
             revert(fmp, 0x44)
         }
     }
@@ -58,8 +58,8 @@ library CustomRevert {
         assembly ("memory-safe") {
             let fmp := mload(0x40)
             mstore(fmp, selector)
-            mstore(add(fmp, 0x04), and(value1, 0xffffffffffffffffffffffffffffffffffffffff))
-            mstore(add(fmp, 0x24), and(value2, 0xffffffffffffffffffffffffffffffffffffffff))
+            mstore(add(fmp, 0x04), value1)
+            mstore(add(fmp, 0x24), value2)
             revert(fmp, 0x44)
         }
     }
@@ -68,8 +68,8 @@ library CustomRevert {
     function revertWith(bytes4 selector, address value1, address value2) internal pure {
         assembly ("memory-safe") {
             mstore(0, selector)
-            mstore(0x04, and(value1, 0xffffffffffffffffffffffffffffffffffffffff))
-            mstore(0x24, and(value2, 0xffffffffffffffffffffffffffffffffffffffff))
+            mstore(0x04, value1)
+            mstore(0x24, value2)
             revert(0, 0x44)
         }
     }
