diff --git a/test/utils/Deployers.sol b/test/utils/Deployers.sol
index 02ac8f49..23fc12c6 100644
--- a/test/utils/Deployers.sol
+++ b/test/utils/Deployers.sol
@@ -36,9 +36,9 @@ contract Deployers {
     uint160 constant SQRT_RATIO_1_4 = Constants.SQRT_RATIO_1_4;
     uint160 constant SQRT_RATIO_4_1 = Constants.SQRT_RATIO_4_1;
 
-    IPoolManager.ModifyLiquidityParams public LIQ_PARAMS =
+    IPoolManager.ModifyLiquidityParams internal LIQ_PARAMS =
         IPoolManager.ModifyLiquidityParams({tickLower: -120, tickUpper: 120, liquidityDelta: 1e18});
-    IPoolManager.ModifyLiquidityParams public REMOVE_LIQ_PARAMS =
+    IPoolManager.ModifyLiquidityParams internal REMOVE_LIQ_PARAMS =
         IPoolManager.ModifyLiquidityParams({tickLower: -120, tickUpper: 120, liquidityDelta: -1e18});
 
     // Global variables
