diff --git a/test/foundry-tests/DynamicFees.t.sol b/test/foundry-tests/DynamicFees.t.sol
index 7441eae7..162e15a6 100644
--- a/test/foundry-tests/DynamicFees.t.sol
+++ b/test/foundry-tests/DynamicFees.t.sol
@@ -32,6 +32,8 @@ contract DynamicFees is IDynamicFeeManager {
 }
 
 contract TestDynamicFees is Test, Deployers {
+    uint160 constant SQRT_RATIO_1_1 = 79228162514264337593543950336;
+
     DynamicFees dynamicFees = DynamicFees(
         address(
             uint160(0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF)
