diff --git a/src/test/EmptyTestHooks.sol b/src/test/EmptyTestHooks.sol
index 117312a8..631a98c5 100644
--- a/src/test/EmptyTestHooks.sol
+++ b/src/test/EmptyTestHooks.sol
@@ -23,7 +23,8 @@ contract EmptyTestHooks is IHooks {
                 afterSwap: true,
                 beforeDonate: true,
                 afterDonate: true,
-                noOp: true
+                noOp: true,
+                accessLock: true
             })
         );
     }
