diff --git a/contracts/token/ERC20.sol b/contracts/token/ERC20.sol
index e8461bc7..4301a636 100644
--- a/contracts/token/ERC20.sol
+++ b/contracts/token/ERC20.sol
@@ -9,7 +9,6 @@ import "../libraries/SafeMath.sol";
 contract ERC20 is IERC20 {
     using SafeMath for uint256;
 
-    // ERC-20 data
     string public name;
     string public symbol;
     uint8 public decimals;
@@ -17,7 +16,7 @@ contract ERC20 is IERC20 {
     mapping (address => uint256) public balanceOf;
     mapping (address => mapping (address => uint256)) public allowance;
 
-    // ERC-191 data
+    // EIP-191
     uint256 public chainId;
     mapping (address => uint) public nonceFor;
 
@@ -35,7 +34,7 @@ contract ERC20 is IERC20 {
         require(chainId == 0, "ERC20: ALREADY_INITIALIZED");
         chainId = _chainId;
     }
-    
+
     function mint(address to, uint256 value) internal {
         totalSupply = totalSupply.add(value);
         balanceOf[to] = balanceOf[to].add(value);
@@ -48,7 +47,7 @@ contract ERC20 is IERC20 {
         emit Transfer(from, to, value);
     }
 
-    function _burn(address from, uint256 value) internal {
+    function _burn(address from, uint256 value) private {
         balanceOf[from] = balanceOf[from].sub(value);
         totalSupply = totalSupply.sub(value);
         emit Transfer(from, address(0), value);
@@ -72,7 +71,7 @@ contract ERC20 is IERC20 {
         return true;
     }
 
-    function burn(uint256 value) external {
+    function burn(uint256 value) public {
         _burn(msg.sender, value);
     }
 
