diff --git a/test/Fees.t.sol b/test/Fees.t.sol
index f99b4b6e..a79457f9 100644
--- a/test/Fees.t.sol
+++ b/test/Fees.t.sol
@@ -178,7 +178,7 @@ contract FeesTest is Test, Deployers, GasSnapshot {
 
         if ((protocolSwapFee0 != 0 && protocolSwapFee0 < 4) || (protocolSwapFee1 != 0 && protocolSwapFee1 < 4)) {
             protocolSwapFee = 0;
-            vm.expectRevert(IFees.ProtocolFeeControllerCallFailedOrInvalidResult.selector);
+            vm.expectRevert(IFees.FeeTooLarge.selector);
         }
         manager.setProtocolFees(key0);
 
@@ -225,7 +225,7 @@ contract FeesTest is Test, Deployers, GasSnapshot {
         ) {
             protocolSwapFee = 0;
             protocolWithdrawFee = 0;
-            vm.expectRevert(IFees.ProtocolFeeControllerCallFailedOrInvalidResult.selector);
+            vm.expectRevert(IFees.FeeTooLarge.selector);
         }
         manager.setProtocolFees(key2);
 
