diff --git a/contracts/libraries/FeeVoting.sol b/contracts/libraries/FeeVoting.sol
index 21c1c260..2cb439dc 100644
--- a/contracts/libraries/FeeVoting.sol
+++ b/contracts/libraries/FeeVoting.sol
@@ -5,17 +5,19 @@ import "./SafeMath.sol";
 
 import "../UniswapV3Pair.sol";
 
-library FeeVoting {
-    using SafeMath for uint;
-    using SafeMath for uint112;
-    using SafeMath for  int;
-    using SafeMath for  int128;
+// functions for aggregate fee votes
 
+library FeeVoting {
     struct Aggregate {
         int128 numerator;
         int128 denominator;
     }
 
+    using SafeMath for uint;
+    using SafeMath for uint112;
+    using SafeMath for  int;
+    using SafeMath for  int128;
+
     function add(Aggregate memory x, Aggregate memory y) internal pure returns (Aggregate memory z) {
         // freshman sum
         z = Aggregate({
