diff --git a/contracts/libraries/FixedPointExtra.sol b/contracts/libraries/FixedPointExtra.sol
index 527cf8df..cb587ae4 100644
--- a/contracts/libraries/FixedPointExtra.sol
+++ b/contracts/libraries/FixedPointExtra.sol
@@ -11,7 +11,7 @@ library FixedPointExtra {
     // reverts on overflow
     function muli(FixedPoint.uq112x112 memory self, int y) internal pure returns (int) {
         uint144 z = FixedPoint.decode144(FixedPoint.mul(self, uint(y < 0 ? -y : y)));
-        return y < 0 ? -int(z) : z;
+        return y < 0 ? -z : z;
     }
 
     // multiply a UQ112x112 by a UQ112x112, returning a UQ112x112
