diff --git a/src/libraries/FixedPoint128.sol b/src/libraries/FixedPoint128.sol
index abf1e3d2..2dbdfc1e 100644
--- a/src/libraries/FixedPoint128.sol
+++ b/src/libraries/FixedPoint128.sol
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
-pragma solidity ^0.8.20;
+pragma solidity ^0.8.0;
 
 /// @title FixedPoint128
 /// @notice A library for handling binary fixed point numbers, see https://en.wikipedia.org/wiki/Q_(number_format)
