diff --git a/contracts/libraries/FixedPointExtra.sol b/contracts/libraries/FixedPointExtra.sol
index cb587ae4..ef5d372c 100644
--- a/contracts/libraries/FixedPointExtra.sol
+++ b/contracts/libraries/FixedPointExtra.sol
@@ -15,6 +15,7 @@ library FixedPointExtra {
     }
 
     // multiply a UQ112x112 by a UQ112x112, returning a UQ112x112
+
     function muluq(FixedPoint.uq112x112 memory self, FixedPoint.uq112x112 memory other)
         internal
         pure
