diff --git a/contracts/libraries/FullMath.sol b/contracts/libraries/FullMath.sol
index b81489d4..89dc917d 100644
--- a/contracts/libraries/FullMath.sol
+++ b/contracts/libraries/FullMath.sol
@@ -12,7 +12,7 @@ library FullMath {
     /// @return l The least significant portion of an emulated 512 bit width integer
     /// @return h The most significant portion of an emulated 512 bit width integer
     function fullMul(uint256 x, uint256 y) private pure returns (uint256 l, uint256 h) {
-        uint256 mm = mulmod(x, y, uint256(-1));
+        uint256 mm = mulmod(x, y, type(uint256).max);
         l = x * y;
         h = mm - l;
         if (mm < l) h -= 1;
