diff --git a/contracts/test/FullMathEchidnaTest.sol b/contracts/test/FullMathEchidnaTest.sol
index 4f1b66a8..f68d8ec3 100644
--- a/contracts/test/FullMathEchidnaTest.sol
+++ b/contracts/test/FullMathEchidnaTest.sol
@@ -27,4 +27,26 @@ contract FullMathEchidnaTest {
             assert(ceiled == floored);
         }
     }
+
+    function checkMulDiv(
+        uint256 x,
+        uint256 y,
+        uint256 d
+    ) external pure {
+        require(d > 0);
+        uint256 z = FullMath.mulDiv(x, y, d);
+        if (x == 0 || y == 0) {
+            assert(z == 0);
+            return;
+        }
+
+        // recompute x and y via mulDiv of the result of floor(x*y/d), should always be less than original inputs by < d
+        uint256 x2 = FullMath.mulDiv(z, d, y);
+        uint256 y2 = FullMath.mulDiv(z, d, x);
+        assert(x2 <= x);
+        assert(y2 <= y);
+
+        assert(x - x2 < d);
+        assert(y - y2 < d);
+    }
 }
