diff --git a/contracts/test/FullMathEchidnaTest.sol b/contracts/test/FullMathEchidnaTest.sol
index 3b3f8283..51c69bc0 100644
--- a/contracts/test/FullMathEchidnaTest.sol
+++ b/contracts/test/FullMathEchidnaTest.sol
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity =0.8.19;
 
-import {FullMath} from "../libraries/FullMath.sol";
+import {FullMath} from '../libraries/FullMath.sol';
 
 contract FullMathEchidnaTest {
     function checkMulDivRounding(uint256 x, uint256 y, uint256 d) external pure {
