diff --git a/test/Hooks.t.sol b/test/Hooks.t.sol
index db070764..9bf1bb0c 100644
--- a/test/Hooks.t.sol
+++ b/test/Hooks.t.sol
@@ -87,7 +87,7 @@ contract HooksTest is Test, Deployers, GasSnapshot {
             IPoolManager.SwapParams({zeroForOne: true, amountSpecified: 100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
 
         PoolSwapTest.TestSettings memory testSettings =
-            PoolSwapTest.TestSettings({withdrawTokens: true, settleUsingTransfer: true, currencyAlreadySent: false});
+            PoolSwapTest.TestSettings({withdrawTokens: true, settleUsingTransfer: true});
 
         swapRouter.swap(key, swapParams, testSettings, new bytes(222));
         assertEq(mockHooks.beforeSwapData(), new bytes(222));
@@ -100,7 +100,7 @@ contract HooksTest is Test, Deployers, GasSnapshot {
         swapRouter.swap(
             key,
             IPoolManager.SwapParams(false, 100, SQRT_RATIO_1_1 + 60),
-            PoolSwapTest.TestSettings(false, false, false),
+            PoolSwapTest.TestSettings(false, false),
             ZERO_BYTES
         );
     }
@@ -111,7 +111,7 @@ contract HooksTest is Test, Deployers, GasSnapshot {
         swapRouter.swap(
             key,
             IPoolManager.SwapParams(false, 100, SQRT_RATIO_1_1 + 60),
-            PoolSwapTest.TestSettings(false, false, false),
+            PoolSwapTest.TestSettings(false, false),
             ZERO_BYTES
         );
     }
