diff --git a/src/interfaces/external/IERC20Minimal.sol b/src/interfaces/external/IERC20Minimal.sol
index 9df10873..c63aefca 100644
--- a/src/interfaces/external/IERC20Minimal.sol
+++ b/src/interfaces/external/IERC20Minimal.sol
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
-pragma solidity ^0.8.20;
+pragma solidity ^0.8.0;
 
 /// @title Minimal ERC20 interface for Uniswap
 /// @notice Contains a subset of the full ERC20 interface that is used in Uniswap V3
