diff --git a/contracts/interfaces/IERC20.sol b/contracts/interfaces/IERC20.sol
index 40d67d10..29a6f810 100644
--- a/contracts/interfaces/IERC20.sol
+++ b/contracts/interfaces/IERC20.sol
@@ -13,6 +13,7 @@ interface IERC20 {
 
     function DOMAIN_SEPARATOR() external view returns (bytes32);
     function APPROVE_TYPEHASH() external pure returns (bytes32);
+    function chainId() external returns (uint256);
     function nonces(address owner) external view returns (uint256);
 
     function transfer(address to, uint256 value) external returns (bool);
