diff --git a/contracts/interfaces/IERC20.sol b/contracts/interfaces/IERC20.sol
index b19f43be..ce8d093e 100644
--- a/contracts/interfaces/IERC20.sol
+++ b/contracts/interfaces/IERC20.sol
@@ -16,10 +16,10 @@ interface IERC20 {
     function nonces(address owner) external view returns (uint);
 
     function transfer(address to, uint value) external returns (bool);
-    function forfeit(uint value) external;
+    function unsafeBurn(uint value) external;
     function approve(address spender, uint value) external returns (bool);
     function transferFrom(address from, address to, uint value) external returns (bool);
-    function forfeitFrom(address from, uint value) external;
+    function unsafeBurnFrom(address from, uint value) external;
     function permit(
         address owner, address spender, uint value, uint nonce, uint expiration, uint8 v, bytes32 r, bytes32 s
     )
