diff --git a/src/interfaces/IHooks.sol b/src/interfaces/IHooks.sol
index 2e64acb3..34efe9da 100644
--- a/src/interfaces/IHooks.sol
+++ b/src/interfaces/IHooks.sol
@@ -97,7 +97,7 @@ interface IHooks {
     /// @param params The parameters for the swap
     /// @param hookData Arbitrary data handed into the PoolManager by the swapper to be be passed on to the hook
     /// @return bytes4 The function selector for the hook
-    /// @return int128 The hook's delta in specified currency. Positive: the hook is owed/took currency, negative: the hook owes/sent currency
+    /// @return int256 The hook's delta in specified currency. Positive: the hook is owed/took currency, negative: the hook owes/sent currency
     function beforeSwap(
         address sender,
         PoolKey calldata key,
