diff --git a/src/interfaces/IPoolManager.sol b/src/interfaces/IPoolManager.sol
index 5fd8b4a7..1c891640 100644
--- a/src/interfaces/IPoolManager.sol
+++ b/src/interfaces/IPoolManager.sol
@@ -158,11 +158,10 @@ interface IPoolManager is IProtocolFees, IERC6909Claims, IExtsload {
     /// @param key The pool to modify liquidity in
     /// @param params The parameters for modifying the liquidity
     /// @param hookData Any data to pass to the callback, via `IUnlockCallback(msg.sender).unlockCallback(data)`
-    /// @return delta The balance delta of the liquidity change
-    /// @return feeDelta The balance delta of the fees generated in the liquidity range
+    /// @return delta The balance delta of the liquidity
     function modifyLiquidity(PoolKey memory key, ModifyLiquidityParams memory params, bytes calldata hookData)
         external
-        returns (BalanceDelta, BalanceDelta);
+        returns (BalanceDelta);
 
     struct SwapParams {
         bool zeroForOne;
