diff --git a/contracts/interfaces/IPoolManager.sol b/contracts/interfaces/IPoolManager.sol
index f7fb9a5c..6eb73e9c 100644
--- a/contracts/interfaces/IPoolManager.sol
+++ b/contracts/interfaces/IPoolManager.sol
@@ -3,10 +3,9 @@ pragma solidity >=0.6.2;
 
 import {IERC20Minimal} from './external/IERC20Minimal.sol';
 import {Pool} from '../libraries/Pool.sol';
-import {IERC1155} from '@openzeppelin/contracts/token/ERC1155/IERC1155.sol';
 import {IHooks} from './IHooks.sol';
 
-interface IPoolManager is IERC1155 {
+interface IPoolManager {
     /// @notice Thrown when tokens touched has exceeded max of 256
     error MaxTokensTouched();
 
@@ -19,9 +18,6 @@ interface IPoolManager is IERC1155 {
     /// @param locker The current locker
     error LockedBy(address locker);
 
-    /// @notice The ERC1155 being deposited is not the Uniswap ERC1155
-    error NotPoolManagerToken();
-
     /// @notice Returns the key for identifying a pool
     struct PoolKey {
         /// @notice The lower token of the pool, sorted numerically
@@ -112,13 +108,6 @@ interface IPoolManager is IERC1155 {
         uint256 amount
     ) external;
 
-    /// @notice Called by the user to move value into ERC1155 balance
-    function mint(
-        IERC20Minimal token,
-        address to,
-        uint256 amount
-    ) external;
-
     /// @notice Called by the user to pay what is owed
     function settle(IERC20Minimal token) external returns (uint256 paid);
 
