diff --git a/contracts/interfaces/pair/IUniswapV3PairDerivedState.sol b/contracts/interfaces/pair/IUniswapV3PairDerivedState.sol
index 7da5679b..30600ee3 100644
--- a/contracts/interfaces/pair/IUniswapV3PairDerivedState.sol
+++ b/contracts/interfaces/pair/IUniswapV3PairDerivedState.sol
@@ -26,5 +26,5 @@ interface IUniswapV3PairDerivedState {
     /// @return tickCumulative Cumulative tick value as of `secondsAgo` from the current block timestamp
     /// @return liquidityCumulative Cumulative liquidity-in-range value as of `secondsAgo` from the current block
     /// timestamp
-    function observe(uint32 secondsAgo) external view returns (int56 tickCumulative, uint160 liquidityCumulative);
+    function scry(uint32 secondsAgo) external view returns (int56 tickCumulative, uint160 liquidityCumulative);
 }
