diff --git a/contracts/interfaces/IUniswapV2.sol b/contracts/interfaces/IUniswapV2.sol
index 60a11722..1b247fe2 100644
--- a/contracts/interfaces/IUniswapV2.sol
+++ b/contracts/interfaces/IUniswapV2.sol
@@ -28,7 +28,7 @@ interface IUniswapV2 {
         uint128 reserve1,
         address input
     );
-    event FeeLiquidityMinted(uint liquidity);
+    event FeesMinted(uint liquidity);
 
     function factory() external view returns (address);
     function token0() external view returns (address);
