diff --git a/contracts/interfaces/pair/IUniswapV3PairState.sol b/contracts/interfaces/pair/IUniswapV3PairState.sol
index c3fdbfde..03c4646e 100644
--- a/contracts/interfaces/pair/IUniswapV3PairState.sol
+++ b/contracts/interfaces/pair/IUniswapV3PairState.sol
@@ -16,7 +16,6 @@ interface IUniswapV3PairState {
         );
 
     function feeGrowthGlobal0X128() external view returns (uint256);
-
     function feeGrowthGlobal1X128() external view returns (uint256);
 
     function protocolFees() external view returns (uint128 token0, uint128 token1);
