diff --git a/contracts/interfaces/IUniswapV2Factory.sol b/contracts/interfaces/IUniswapV2Factory.sol
index d7088662..ad6eab6e 100644
--- a/contracts/interfaces/IUniswapV2Factory.sol
+++ b/contracts/interfaces/IUniswapV2Factory.sol
@@ -10,7 +10,6 @@ interface IUniswapV2Factory {
     function getTokens(address exchange) external view returns (address, address);
     function getExchange(address tokenA, address tokenB) external view returns (address);
     function getOtherTokens(address token) external view returns (address[] memory);
-    function getOtherTokensLength(address token) external view returns (uint256);
 
     function createExchange(address tokenA, address tokenB) external returns (address exchange);
 }
