diff --git a/contracts/interfaces/IUniswapV2.sol b/contracts/interfaces/IUniswapV2.sol
index a0126048..c7dcd47d 100644
--- a/contracts/interfaces/IUniswapV2.sol
+++ b/contracts/interfaces/IUniswapV2.sol
@@ -9,6 +9,7 @@ interface IUniswapV2 {
     function factory() external view returns (address);
     function token0() external view returns (address);
     function token1() external view returns (address);
+
     function reserve0() external view returns (uint112);
     function reserve1() external view returns (uint112);
     function blockNumberLast() external view returns (uint32);
@@ -18,6 +19,7 @@ interface IUniswapV2 {
     function make() external returns (uint liquidity);
     function made() external returns (uint amount0, uint amount1);
     function move(address tokenIn, uint amountOut) external;
+
     function skim() external;
     function sync() external;
 
