diff --git a/contracts/interfaces/IUniswapV3Pair.sol b/contracts/interfaces/IUniswapV3Pair.sol
index 073132a3..df3a6e1f 100644
--- a/contracts/interfaces/IUniswapV3Pair.sol
+++ b/contracts/interfaces/IUniswapV3Pair.sol
@@ -1,6 +1,10 @@
 pragma solidity >=0.6.2;
 
-interface IUniswapV3Pair {
+import "./IUniswapV3ERC20.sol";
+
+interface IUniswapV3Pair is IUniswapV3ERC20 {
+    event Mint(address indexed sender, uint amount0, uint amount1);
+    event Burn(address indexed sender, uint amount0, uint amount1, address indexed to);
     event Swap(
         address indexed sender,
         bool tokenIn,
@@ -8,11 +12,10 @@ interface IUniswapV3Pair {
         uint amountOut,
         address indexed to
     );
+    event Give(address indexed from, uint liquidity);
     event Shift(uint16 tick);
-    event Add(address indexed sender, uint112 liquidity, uint16 lowerTick, uint16 upperTick);
-    event Remove(address indexed sender, uint112 liquidity, uint16 lowerTick, uint16 upperTick);
 
-    function MINIMUM_LIQUIDITY() external pure returns (uint112);
+    function MINIMUM_LIQUIDITY() external pure returns (uint);
     function factory() external view returns (address);
     function token0() external view returns (address);
     function token1() external view returns (address);
@@ -21,7 +24,9 @@ interface IUniswapV3Pair {
     function price1CumulativeLast() external view returns (uint);
     function kLast() external view returns (uint);
 
-    function initialAdd(uint112 amount0, uint112 amount1, uint16 startingTick) external returns (uint112 liquidity);
-    function add(uint112 liquidity, uint16 lowerTick, uint16 upperTick) external;
-    function remove(uint112 liquidity, uint16 lowerTick, uint16 upperTick) external;
+    function mint(address to) external returns (uint liquidity);
+    function burn(address to) external returns (uint amount0, uint amount1);
+//    function swap(uint amount0Out, uint amount1Out, address to, bytes calldata data) external;
+//    function skim(address to) external;
+//    function sync() external;
 }
