diff --git a/contracts/interfaces/IUniswapV3Pair.sol b/contracts/interfaces/IUniswapV3Pair.sol
index bf042a43..175c94be 100644
--- a/contracts/interfaces/IUniswapV3Pair.sol
+++ b/contracts/interfaces/IUniswapV3Pair.sol
@@ -34,7 +34,7 @@ interface IUniswapV3Pair {
             uint8 unlockedAndPriceBit
         );
 
-    function liquidityCurrent() external view returns (uint128);
+    function liquidityCurrent() external view returns (uint128 liquidityCurrent);
 
     function tickBitmap(int16) external view returns (uint256);
 
