diff --git a/contracts/interfaces/IUniswapV3Pair.sol b/contracts/interfaces/IUniswapV3Pair.sol
index d1ef0e67..a3ff6c5c 100644
--- a/contracts/interfaces/IUniswapV3Pair.sol
+++ b/contracts/interfaces/IUniswapV3Pair.sol
@@ -13,5 +13,5 @@ interface IUniswapV3Pair {
     function blockTimestampLast() external view returns (uint32);
 
     function tickCurrent() external view returns (int16);
-    function virtualSupplies(uint) external view returns (uint112);
+    function liquidityVirtuals(uint) external view returns (uint112);
 }
