diff --git a/contracts/interfaces/IUniswapV3Pair.sol b/contracts/interfaces/IUniswapV3Pair.sol
index bfc6b8a7..087f493c 100644
--- a/contracts/interfaces/IUniswapV3Pair.sol
+++ b/contracts/interfaces/IUniswapV3Pair.sol
@@ -1,12 +1,12 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity >=0.5.0;
 
-import './pair/IUniswapV3PairImmutables.sol';
-import './pair/IUniswapV3PairEvents.sol';
-import './pair/IUniswapV3PairActions.sol';
-import './pair/IUniswapV3PairOwnerActions.sol';
-import './pair/IUniswapV3PairState.sol';
-import './pair/IUniswapV3PairDerivedState.sol';
+import './IUniswapV3PairImmutables.sol';
+import './IUniswapV3PairEvents.sol';
+import './IUniswapV3PairActions.sol';
+import './IUniswapV3PairOwnerActions.sol';
+import './IUniswapV3PairState.sol';
+import './IUniswapV3PairDerivedState.sol';
 
 interface IUniswapV3Pair is
     IUniswapV3PairImmutables,
