diff --git a/contracts/interfaces/pair/IUniswapV3PairActions.sol b/contracts/interfaces/pair/IUniswapV3PairActions.sol
index 74424bb2..5549c81d 100644
--- a/contracts/interfaces/pair/IUniswapV3PairActions.sol
+++ b/contracts/interfaces/pair/IUniswapV3PairActions.sol
@@ -5,13 +5,6 @@ interface IUniswapV3PairActions {
     // initialize the pair
     function initialize(uint160 sqrtPriceX96) external;
 
-    // update a position's fees owed to account protocol fees
-    function poke(
-        address owner,
-        int24 tickLower,
-        int24 tickUpper
-    ) external;
-
     // mint some liquidity to an address
     function mint(
         address recipient,
