diff --git a/contracts/interfaces/IUniswapV2Pair.sol b/contracts/interfaces/IUniswapV2Pair.sol
index a3b7003e..5bb8c0d1 100644
--- a/contracts/interfaces/IUniswapV2Pair.sol
+++ b/contracts/interfaces/IUniswapV2Pair.sol
@@ -29,6 +29,4 @@ interface IUniswapV2Pair is IUniswapV2ERC20 {
     function swap(uint amount0Out, uint amount1Out, address to, bytes calldata data) external;
     function skim(address to) external;
     function sync() external;
-
-    function initializeSymbol(string calldata) external;
 }
