diff --git a/contracts/interfaces/IUniswapV2Factory.sol b/contracts/interfaces/IUniswapV2Factory.sol
index e16409e4..5e0923bc 100644
--- a/contracts/interfaces/IUniswapV2Factory.sol
+++ b/contracts/interfaces/IUniswapV2Factory.sol
@@ -1,7 +1,7 @@
 pragma solidity 0.5.14;
 
 interface IUniswapV2Factory {
-    event ExchangeCreated(address indexed token0, address indexed token1, address exchange, uint256);
+    event ExchangeCreated(address indexed token0, address indexed token1, address exchange, uint256 exchangeNumber);
 
     function exchangeBytecode() external view returns (bytes memory);
     function factoryOwner() external view returns (address);
