diff --git a/contracts/interfaces/IUniswapV3Pair.sol b/contracts/interfaces/IUniswapV3Pair.sol
index 923883a8..f4c53084 100644
--- a/contracts/interfaces/IUniswapV3Pair.sol
+++ b/contracts/interfaces/IUniswapV3Pair.sol
@@ -19,7 +19,7 @@ interface IUniswapV3Pair {
     // variables/state
     function feeTo() external view returns (address);
 
-    function blockTimestampLast() external view returns (uint32);
+    function blockTimestampLast() external view returns (uint64);
 
     function liquidityCurrent() external view returns (uint128);
 
