diff --git a/contracts/interfaces/pair/IUniswapV3PairState.sol b/contracts/interfaces/pair/IUniswapV3PairState.sol
index f57e1118..d7b23a06 100644
--- a/contracts/interfaces/pair/IUniswapV3PairState.sol
+++ b/contracts/interfaces/pair/IUniswapV3PairState.sol
@@ -78,7 +78,7 @@ interface IUniswapV3PairState {
     /// @notice Returns data about a specific observation index
     /// @param index the element of the observations array to fetch
     /// @dev You most likely want to use #scry instead of this method to get an observation as of some amount of time
-    ///      ago rather than at a specific index in the array
+    ///     ago rather than at a specific index in the array
     /// @return blockTimestamp the timestamp of the observation
     function observations(uint256 index)
         external
