diff --git a/contracts/interfaces/IUniswapV3Pair.sol b/contracts/interfaces/IUniswapV3Pair.sol
index 53ce790f..be32680e 100644
--- a/contracts/interfaces/IUniswapV3Pair.sol
+++ b/contracts/interfaces/IUniswapV3Pair.sol
@@ -16,12 +16,6 @@ interface IUniswapV3Pair {
 
     function fee() external pure returns (uint24);
 
-    function tickSpacing() external pure returns (int24);
-
-    function MIN_TICK() external pure returns (int24);
-
-    function MAX_TICK() external pure returns (int24);
-
     // variables/state
     function feeTo() external view returns (address);
 
@@ -48,8 +42,6 @@ interface IUniswapV3Pair {
     // derived state
     function isInitialized() external view returns (bool);
 
-    function tickCurrent() external view returns (int24);
-
     function getCumulatives()
         external
         view
