diff --git a/contracts/interfaces/IUniswapV3Pair.sol b/contracts/interfaces/IUniswapV3Pair.sol
index f4c53084..6e6ecbc7 100644
--- a/contracts/interfaces/IUniswapV3Pair.sol
+++ b/contracts/interfaces/IUniswapV3Pair.sol
@@ -14,7 +14,7 @@ interface IUniswapV3Pair {
 
     function token1() external pure returns (address);
 
-    function fee() external pure returns (uint24);
+    function fee() external pure returns (uint16);
 
     // variables/state
     function feeTo() external view returns (address);
