diff --git a/contracts/interfaces/IUniswapV2Factory.sol b/contracts/interfaces/IUniswapV2Factory.sol
index 040aa22c..e16409e4 100644
--- a/contracts/interfaces/IUniswapV2Factory.sol
+++ b/contracts/interfaces/IUniswapV2Factory.sol
@@ -1,4 +1,4 @@
-pragma solidity 0.5.15;
+pragma solidity 0.5.14;
 
 interface IUniswapV2Factory {
     event ExchangeCreated(address indexed token0, address indexed token1, address exchange, uint256);
