diff --git a/contracts/interfaces/IUniswapV2.sol b/contracts/interfaces/IUniswapV2.sol
index 3ab2450a..f6be4c8f 100644
--- a/contracts/interfaces/IUniswapV2.sol
+++ b/contracts/interfaces/IUniswapV2.sol
@@ -8,17 +8,19 @@ interface IUniswapV2 {
         address indexed sender, address indexed recipient, uint256 liquidity, uint128 amountToken0, uint128 amountToken1
     );
     event Swap(
-        address indexed sender, address indexed recipient, address input, uint128 amountToken0, uint128 amountToken1
+        address indexed sender, address indexed recipient, address input, uint128 amountInput, uint128 amountOutput
     );
 
     function factory() external view returns (address);
     function token0() external view returns (address);
     function token1() external view returns (address);
 
-    function getReservesCumulativeAndOverflows() external view returns (uint128, uint128, uint128, uint128);
+    function getReserves() external view returns (uint128, uint128);
+    function getReservesCumulative() external view returns (uint128, uint128);
 
-    function getAmountOutput(uint128 amountInput, uint128 reserveInput, uint128 reserveOutput)
-        external pure returns (uint128 amountOutput);
+    function getAmountOutput(
+        uint128 amountInput, uint128 reserveInput, uint128 reserveOutput
+    ) external pure returns (uint128 amountOutput);
 
     function mintLiquidity(address recipient) external returns (uint256 liquidity);
     function burnLiquidity(address recipient) external returns (uint128 amountToken0, uint128 amountToken1);
