diff --git a/contracts/interfaces/pair/IUniswapV3PairImmutables.sol b/contracts/interfaces/pair/IUniswapV3PairImmutables.sol
index 77212dd2..258cbea4 100644
--- a/contracts/interfaces/pair/IUniswapV3PairImmutables.sol
+++ b/contracts/interfaces/pair/IUniswapV3PairImmutables.sol
@@ -5,7 +5,6 @@ interface IUniswapV3PairImmutables {
     function factory() external view returns (address);
 
     function token0() external view returns (address);
-
     function token1() external view returns (address);
 
     function fee() external view returns (uint24);
@@ -13,7 +12,6 @@ interface IUniswapV3PairImmutables {
     function tickSpacing() external view returns (int24);
 
     function minTick() external view returns (int24);
-
     function maxTick() external view returns (int24);
 
     function maxLiquidityPerTick() external view returns (uint128);
