diff --git a/contracts/interfaces/IUniswapV3Pair.sol b/contracts/interfaces/IUniswapV3Pair.sol
index a3a49346..61539ba4 100644
--- a/contracts/interfaces/IUniswapV3Pair.sol
+++ b/contracts/interfaces/IUniswapV3Pair.sol
@@ -42,24 +42,6 @@ interface IUniswapV3Pair {
 
     function getCumulativePrices() external view returns (uint256 price0Cumulative, uint256 price1Cumulative);
 
-    // swapping
-    function swap0For1(
-        uint112 amount0In,
-        address to,
-        bytes calldata data
-    ) external returns (uint112 amount1Out);
-
-    function swap1For0(
-        uint112 amount1In,
-        address to,
-        bytes calldata data
-    ) external returns (uint112 amount0Out);
-
-    // called by feeToSetter of the factory
-    function feeTo() external view returns (address);
-
-    function setFeeTo(address) external;
-
     // allows the factory feeToSetter address to recover any tokens other than token0 and token1 held by the contract
     function recover(
         address token,
