diff --git a/contracts/interfaces/IUniswapV2.sol b/contracts/interfaces/IUniswapV2.sol
index c7dcd47d..106b3ab5 100644
--- a/contracts/interfaces/IUniswapV2.sol
+++ b/contracts/interfaces/IUniswapV2.sol
@@ -1,9 +1,9 @@
 pragma solidity 0.5.14;
 
 interface IUniswapV2 {
-    event Make(address indexed sender, uint amount0, uint amount1);
-    event Made(address indexed sender, uint amount0, uint amount1);
-    event Move(address indexed sender, address indexed tokenIn, uint amountIn, uint amountOut);
+    event Mint(address indexed sender, uint amount0, uint amount1);
+    event Burn(address indexed sender, uint amount0, uint amount1);
+    event Swap(address indexed sender, address indexed tokenIn, uint amountIn, uint amountOut);
     event Sync(uint112 reserve0, uint112 reserve1);
 
     function factory() external view returns (address);
@@ -16,12 +16,12 @@ interface IUniswapV2 {
     function price0CumulativeLast() external view returns (uint);
     function price1CumulativeLast() external view returns (uint);
 
-    function make() external returns (uint liquidity);
-    function made() external returns (uint amount0, uint amount1);
-    function move(address tokenIn, uint amountOut) external;
+    function mint() external returns (uint liquidity);
+    function burn() external returns (uint amount0, uint amount1);
+    function swap(address tokenIn, uint amountOut) external;
 
     function skim() external;
     function sync() external;
 
-    function initialize(address, address) external; // only called once by the factory on deploy
+    function initialize(address _token0, address _token1) external; // only called once by the factory on deploy
 }
