diff --git a/contracts/libraries/LiquidityMath.sol b/contracts/libraries/LiquidityMath.sol
index 58b8bff5..a9e7ece0 100644
--- a/contracts/libraries/LiquidityMath.sol
+++ b/contracts/libraries/LiquidityMath.sol
@@ -22,7 +22,7 @@ library LiquidityMath {
     function addCapped(uint128 x, uint256 y) internal pure returns (uint128 z) {
         z = uint128(x + y);
         if (z < x) {
-            z = uint128(-1);
+            z = type(uint128).max;
         }
     }
 }
