diff --git a/contracts/test/MockTimeUniswapV3Pair.sol b/contracts/test/MockTimeUniswapV3Pair.sol
index d96e2db3..51675596 100644
--- a/contracts/test/MockTimeUniswapV3Pair.sol
+++ b/contracts/test/MockTimeUniswapV3Pair.sol
@@ -2,6 +2,9 @@
 pragma solidity =0.7.6;
 
 import '../UniswapV3Pair.sol';
+import '../UniswapV3PairDeployer.sol';
+
+import '../libraries/Oracle.sol';
 
 // used for testing time dependent behavior
 contract MockTimeUniswapV3Pair is UniswapV3Pair {
