diff --git a/contracts/test/MockContract.sol b/contracts/test/MockContract.sol
index f9863b2b..751189da 100644
--- a/contracts/test/MockContract.sol
+++ b/contracts/test/MockContract.sol
@@ -1,7 +1,6 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity =0.8.19;
-
-import {Proxy} from "@openzeppelin/contracts/proxy/Proxy.sol";
+import {Proxy} from '@openzeppelin/contracts/proxy/Proxy.sol';
 
 /// @notice Mock contract that tracks the number of calls to various functions by selector
 /// @dev allows for proxying to an implementation contract
