diff --git a/src/test/NoOpTestHooks.sol b/src/test/NoOpTestHooks.sol
index b9c10eba..7007ef99 100644
--- a/src/test/NoOpTestHooks.sol
+++ b/src/test/NoOpTestHooks.sol
@@ -22,7 +22,8 @@ contract NoOpTestHooks is BaseTestHooks {
                 afterSwap: false,
                 beforeDonate: true,
                 afterDonate: false,
-                noOp: true
+                noOp: true,
+                accessLock: false
             })
         );
     }
