diff --git a/contracts/test/OracleTest.sol b/contracts/test/OracleTest.sol
index 2cf585d6..d57f2a3f 100644
--- a/contracts/test/OracleTest.sol
+++ b/contracts/test/OracleTest.sol
@@ -37,7 +37,11 @@ contract OracleTest {
         time = _time;
     }
 
-    function scry(uint32 secondsAgo) external view returns (int56 tickCumulative, uint160 liquidityCumulative) {
+    function scry(uint32 secondsAgo)
+        external
+        view
+        returns (int56 tickCumulative, uint160 liquidityCumulative)
+    {
         return observations.scry(time, secondsAgo, tick, index, liquidity);
     }
 }
