diff --git a/test/foundry-tests/Owned.t.sol b/test/foundry-tests/Owned.t.sol
index f567cd7f..3e269c7d 100644
--- a/test/foundry-tests/Owned.t.sol
+++ b/test/foundry-tests/Owned.t.sol
@@ -1,8 +1,8 @@
 pragma solidity ^0.8.19;
 
-import {Test} from "forge-std/Test.sol";
-import {Vm} from "forge-std/Vm.sol";
-import {Owned} from "../../contracts/Owned.sol";
+import {Test} from 'forge-std/Test.sol';
+import {Vm} from 'forge-std/Vm.sol';
+import {Owned} from '../../contracts/Owned.sol';
 
 contract OwnedTest is Test {
     Owned owned;
