diff --git a/src/libraries/ParseBytes.sol b/src/libraries/ParseBytes.sol
index 3429b546..2f6ba217 100644
--- a/src/libraries/ParseBytes.sol
+++ b/src/libraries/ParseBytes.sol
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 
-pragma solidity ^0.8.20;
+pragma solidity ^0.8.0;
 
 /// @notice Parses bytes returned from hooks and the byte selector used to check return selectors from hooks.
 /// @dev parseSelector also is used to parse the expected selector
