diff --git a/src/PoolManager.sol b/src/PoolManager.sol
index 034ff167..3982ea14 100644
--- a/src/PoolManager.sol
+++ b/src/PoolManager.sol
@@ -15,7 +15,7 @@ import {IPoolManager} from "./interfaces/IPoolManager.sol";
 import {IUnlockCallback} from "./interfaces/callback/IUnlockCallback.sol";
 import {ProtocolFees} from "./ProtocolFees.sol";
 import {ERC6909Claims} from "./ERC6909Claims.sol";
-import {PoolId} from "./types/PoolId.sol";
+import {PoolId, PoolIdLibrary} from "./types/PoolId.sol";
 import {BalanceDelta, BalanceDeltaLibrary} from "./types/BalanceDelta.sol";
 import {BeforeSwapDelta} from "./types/BeforeSwapDelta.sol";
 import {Lock} from "./libraries/Lock.sol";
@@ -77,6 +77,7 @@ import {CustomRevert} from "./libraries/CustomRevert.sol";
 /// @title PoolManager
 /// @notice Holds the state for all pools
 contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claims, Extsload, Exttload {
+    using PoolIdLibrary for PoolKey;
     using SafeCast for *;
     using Pool for *;
     using Hooks for IHooks;
