diff --git a/src/test/PoolSwapTest.sol b/src/test/PoolSwapTest.sol
index 284a7471..0a63a939 100644
--- a/src/test/PoolSwapTest.sol
+++ b/src/test/PoolSwapTest.sol
@@ -11,6 +11,7 @@ import {PoolTestBase} from "./PoolTestBase.sol";
 import {CurrencySettler} from "../../test/utils/CurrencySettler.sol";
 
 contract PoolSwapTest is PoolTestBase {
+    using CurrencyLibrary for Currency;
     using CurrencySettler for Currency;
     using Hooks for IHooks;
 
