diff --git a/src/test/PoolSwapTest.sol b/src/test/PoolSwapTest.sol
index 805c3455..c5fe0ab0 100644
--- a/src/test/PoolSwapTest.sol
+++ b/src/test/PoolSwapTest.sol
@@ -31,7 +31,6 @@ contract PoolSwapTest is Test, PoolTestBase {
     struct TestSettings {
         bool withdrawTokens;
         bool settleUsingTransfer;
-        bool currencyAlreadySent;
     }
 
     function swap(
@@ -41,15 +40,14 @@ contract PoolSwapTest is Test, PoolTestBase {
         bytes memory hookData
     ) external payable returns (BalanceDelta delta) {
         delta = abi.decode(
-            manager.lock(address(this), abi.encode(CallbackData(msg.sender, testSettings, key, params, hookData))),
-            (BalanceDelta)
+            manager.lock(abi.encode(CallbackData(msg.sender, testSettings, key, params, hookData))), (BalanceDelta)
         );
 
         uint256 ethBalance = address(this).balance;
         if (ethBalance > 0) CurrencyLibrary.NATIVE.transfer(msg.sender, ethBalance);
     }
 
-    function lockAcquired(address, bytes calldata rawData) external returns (bytes memory) {
+    function lockAcquired(bytes calldata rawData) external returns (bytes memory) {
         require(msg.sender == address(manager));
 
         CallbackData memory data = abi.decode(rawData, (CallbackData));
@@ -106,18 +104,10 @@ contract PoolSwapTest is Test, PoolTestBase {
         }
 
         if (deltaAfter0 > 0) {
-            if (data.testSettings.currencyAlreadySent) {
-                manager.settle(data.key.currency0);
-            } else {
-                _settle(data.key.currency0, data.sender, int128(deltaAfter0), data.testSettings.settleUsingTransfer);
-            }
+            _settle(data.key.currency0, data.sender, int128(deltaAfter0), data.testSettings.settleUsingTransfer);
         }
         if (deltaAfter1 > 0) {
-            if (data.testSettings.currencyAlreadySent) {
-                manager.settle(data.key.currency1);
-            } else {
-                _settle(data.key.currency1, data.sender, int128(deltaAfter1), data.testSettings.settleUsingTransfer);
-            }
+            _settle(data.key.currency1, data.sender, int128(deltaAfter1), data.testSettings.settleUsingTransfer);
         }
         if (deltaAfter0 < 0) {
             _take(data.key.currency0, data.sender, int128(deltaAfter0), data.testSettings.withdrawTokens);
