diff --git a/contracts/test/PoolModifyPositionTest.sol b/contracts/test/PoolModifyPositionTest.sol
index f4b393eb..447c9b85 100644
--- a/contracts/test/PoolModifyPositionTest.sol
+++ b/contracts/test/PoolModifyPositionTest.sol
@@ -36,7 +36,7 @@ contract PoolModifyPositionTest is ILockCallback {
         }
     }
 
-    function lockAcquired(bytes calldata rawData) external returns (bytes memory) {
+    function lockAcquired(uint256, bytes calldata rawData) external returns (bytes memory) {
         require(msg.sender == address(manager));
 
         CallbackData memory data = abi.decode(rawData, (CallbackData));
