diff --git a/contracts/test/PoolSwapTest.sol b/contracts/test/PoolSwapTest.sol
index 18a898bf..e094815e 100644
--- a/contracts/test/PoolSwapTest.sol
+++ b/contracts/test/PoolSwapTest.sol
@@ -43,7 +43,7 @@ contract PoolSwapTest is ILockCallback {
         }
     }
 
-    function lockAcquired(bytes calldata rawData) external returns (bytes memory) {
+    function lockAcquired(uint256, bytes calldata rawData) external returns (bytes memory) {
         require(msg.sender == address(manager));
 
         CallbackData memory data = abi.decode(rawData, (CallbackData));
