diff --git a/src/test/PoolClaimsTest.sol b/src/test/PoolClaimsTest.sol
index 4544918f..d4aa8e04 100644
--- a/src/test/PoolClaimsTest.sol
+++ b/src/test/PoolClaimsTest.sol
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.24;
 
-import {Currency} from "../types/Currency.sol";
+import {Currency, CurrencyLibrary} from "../types/Currency.sol";
 import {BalanceDelta, toBalanceDelta} from "../types/BalanceDelta.sol";
 import {IPoolManager} from "../interfaces/IPoolManager.sol";
 import {PoolKey} from "../types/PoolKey.sol";
@@ -10,6 +10,7 @@ import {SafeCast} from "../libraries/SafeCast.sol";
 import {CurrencySettler} from "../../test/utils/CurrencySettler.sol";
 
 contract PoolClaimsTest is PoolTestBase {
+    using CurrencyLibrary for Currency;
     using CurrencySettler for Currency;
     using SafeCast for uint256;
 
