diff --git a/contracts/libraries/Pool.sol b/contracts/libraries/Pool.sol
index eef14b87..a0de6cd1 100644
--- a/contracts/libraries/Pool.sol
+++ b/contracts/libraries/Pool.sol
@@ -11,6 +11,7 @@ import {FixedPoint128} from './FixedPoint128.sol';
 import {TickMath} from './TickMath.sol';
 import {SqrtPriceMath} from './SqrtPriceMath.sol';
 import {SwapMath} from './SwapMath.sol';
+import {IHooks} from '../interfaces/IHooks.sol';
 import {IPoolManager} from '../interfaces/IPoolManager.sol';
 
 library Pool {
