diff --git a/src/libraries/Pool.sol b/src/libraries/Pool.sol
index cb447be5..2568602d 100644
--- a/src/libraries/Pool.sol
+++ b/src/libraries/Pool.sol
@@ -381,7 +381,6 @@ library Pool {
                 unchecked {
                     // step.amountIn does not include the swap fee, as it's already been taken from it,
                     // so add it back to get the total amountIn and use that to calculate the amount of fees owed to the protocol
-                    // this line cannot overflow due to limits on the size of protocolFee and params.amountSpecified
                     uint256 delta = (step.amountIn + step.feeAmount) * protocolFee / ProtocolFeeLibrary.PIPS_DENOMINATOR;
                     // subtract it from the total fee and add it to the protocol fee
                     step.feeAmount -= delta;
