diff --git a/src/PoolManager.sol b/src/PoolManager.sol
index d81b3a83..30fb40dc 100644
--- a/src/PoolManager.sol
+++ b/src/PoolManager.sol
@@ -157,10 +157,12 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         int256 current = currency.getDelta(msg.sender);
         int256 next = current + delta;
 
-        if (next == 0) {
-            NonZeroDeltaCount.decrement();
-        } else if (current == 0) {
-            NonZeroDeltaCount.increment();
+        unchecked {
+            if (next == 0) {
+                NonZeroDeltaCount.decrement();
+            } else if (current == 0) {
+                NonZeroDeltaCount.increment();
+            }
         }
 
         currency.setDelta(msg.sender, next);
@@ -263,11 +265,9 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
 
     /// @inheritdoc IPoolManager
     function take(Currency currency, address to, uint256 amount) external override onlyWhenUnlocked {
-        unchecked {
-            // subtraction must be safe
-            _accountDelta(currency, -(amount.toInt128()));
-            currency.transfer(to, amount);
-        }
+        // subtraction must be safe
+        _accountDelta(currency, -(amount.toInt128()));
+        currency.transfer(to, amount);
     }
 
     /// @inheritdoc IPoolManager
@@ -285,11 +285,9 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
 
     /// @inheritdoc IPoolManager
     function mint(address to, uint256 id, uint256 amount) external override onlyWhenUnlocked {
-        unchecked {
-            // subtraction must be safe
-            _accountDelta(CurrencyLibrary.fromId(id), -(amount.toInt128()));
-            _mint(to, id, amount);
-        }
+        // subtraction must be safe
+        _accountDelta(CurrencyLibrary.fromId(id), -(amount.toInt128()));
+        _mint(to, id, amount);
     }
 
     /// @inheritdoc IPoolManager
