diff --git a/src/test/PoolModifyPositionTest.sol b/src/test/PoolModifyPositionTest.sol
index c3b7a236..a25925f2 100644
--- a/src/test/PoolModifyPositionTest.sol
+++ b/src/test/PoolModifyPositionTest.sol
@@ -30,9 +30,7 @@ contract PoolModifyPositionTest is Test, PoolTestBase {
         payable
         returns (BalanceDelta delta)
     {
-        delta = abi.decode(
-            manager.lock(address(this), abi.encode(CallbackData(msg.sender, key, params, hookData))), (BalanceDelta)
-        );
+        delta = abi.decode(manager.lock(abi.encode(CallbackData(msg.sender, key, params, hookData))), (BalanceDelta));
 
         uint256 ethBalance = address(this).balance;
         if (ethBalance > 0) {
@@ -40,7 +38,7 @@ contract PoolModifyPositionTest is Test, PoolTestBase {
         }
     }
 
-    function lockAcquired(address, bytes calldata rawData) external returns (bytes memory) {
+    function lockAcquired(bytes calldata rawData) external returns (bytes memory) {
         require(msg.sender == address(manager));
 
         CallbackData memory data = abi.decode(rawData, (CallbackData));
