diff --git a/contracts/PoolManager.sol b/contracts/PoolManager.sol
index 3f723dd2..612b74cc 100644
--- a/contracts/PoolManager.sol
+++ b/contracts/PoolManager.sol
@@ -106,7 +106,7 @@ contract PoolManager is IPoolManager, NoDelegateCall {
     function _accountDelta(IERC20Minimal token, int256 delta) internal {
         if (delta == 0) return;
         _addTokenToSet(token);
-        tokenDelta[token].delta += delta.toInt248();
+        tokenDelta[token].delta += int248(delta);
     }
 
     /// @dev Accumulates a balance change to a map of token to balance changes
