diff --git a/src/test/PoolDonateTest.sol b/src/test/PoolDonateTest.sol
index 5903c812..2bf4aefc 100644
--- a/src/test/PoolDonateTest.sol
+++ b/src/test/PoolDonateTest.sol
@@ -59,9 +59,16 @@ contract PoolDonateTest is PoolTestBase, Test {
 
         assertEq(reserveBefore0, reserveAfter0);
         assertEq(reserveBefore1, reserveAfter1);
+        if (!data.key.hooks.hasPermission(Hooks.NO_OP_FLAG)) {
+            assertEq(deltaAfter0, int256(data.amount0));
+            assertEq(deltaAfter1, int256(data.amount1));
+        }
 
-        assertEq(deltaAfter0, int256(data.amount0));
-        assertEq(deltaAfter1, int256(data.amount1));
+        if (delta == BalanceDeltaLibrary.MAXIMUM_DELTA) {
+            // Check that this hook is allowed to NoOp, then we can return as we dont need to settle
+            assertTrue(data.key.hooks.hasPermission(Hooks.NO_OP_FLAG), "Invalid NoOp returned");
+            return abi.encode(delta);
+        }
 
         if (deltaAfter0 > 0) _settle(data.key.currency0, data.sender, int128(deltaAfter0), true);
         if (deltaAfter1 > 0) _settle(data.key.currency1, data.sender, int128(deltaAfter1), true);
