diff --git a/contracts/test/PoolLockTest.sol b/contracts/test/PoolLockTest.sol
index 3f0133e4..16b2d965 100644
--- a/contracts/test/PoolLockTest.sol
+++ b/contracts/test/PoolLockTest.sol
@@ -5,7 +5,7 @@ import {IPoolManager} from "../interfaces/IPoolManager.sol";
 import {ILockCallback} from "../interfaces/callback/ILockCallback.sol";
 
 contract PoolLockTest is ILockCallback {
-    event LockAcquired();
+    event LockAcquired(uint256 id);
 
     IPoolManager manager;
 
@@ -18,8 +18,8 @@ contract PoolLockTest is ILockCallback {
     }
 
     /// @notice Called by the pool manager on `msg.sender` when a lock is acquired
-    function lockAcquired(bytes calldata) external override returns (bytes memory) {
-        emit LockAcquired();
+    function lockAcquired(uint256 id, bytes calldata) external override returns (bytes memory) {
+        emit LockAcquired(id);
         return "";
     }
 }
