diff --git a/src/PoolManager.sol b/src/PoolManager.sol
index 7178ab39..0cabcd42 100644
--- a/src/PoolManager.sol
+++ b/src/PoolManager.sol
@@ -298,10 +298,7 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         // Because input is `uint256`, only positive amounts can be cleared.
         int128 amountDelta = amount.toInt128();
         if (amountDelta != current) MustClearExactPositiveDelta.selector.revertWith();
-        // negation must be safe as amountDelta is positive
-        unchecked {
-            _accountDelta(currency, -(amountDelta), msg.sender);
-        }
+        _accountDelta(currency, -(amountDelta), msg.sender);
     }
 
     /// @inheritdoc IPoolManager
