diff --git a/src/test/PoolInitializeTest.sol b/src/test/PoolInitializeTest.sol
index b7bf28c8..6a0a0385 100644
--- a/src/test/PoolInitializeTest.sol
+++ b/src/test/PoolInitializeTest.sol
@@ -44,9 +44,17 @@ contract PoolInitializeTest is Test, PoolTestBase {
         int256 delta1 = manager.currencyDelta(address(this), data.key.currency1);
         uint256 nonZeroDC = manager.getLockNonzeroDeltaCount();
 
-        assertEq(delta0, 0, "delta0");
-        assertEq(delta1, 0, "delta1");
-        assertEq(nonZeroDC, 0, "NonzeroDeltaCount");
+        if (!data.key.hooks.hasPermission(Hooks.ACCESS_LOCK_FLAG)) {
+            assertEq(delta0, 0, "delta0");
+            assertEq(delta1, 0, "delta1");
+            assertEq(nonZeroDC, 0, "NonzeroDeltaCount");
+        } else {
+            // settle deltas
+            if (delta0 > 0) _settle(data.key.currency0, data.sender, int128(delta0), true);
+            if (delta1 > 0) _settle(data.key.currency1, data.sender, int128(delta1), true);
+            if (delta0 < 0) _take(data.key.currency0, data.sender, int128(delta0), true);
+            if (delta1 < 0) _take(data.key.currency1, data.sender, int128(delta1), true);
+        }
 
         return abi.encode(tick);
     }
