diff --git a/src/test/PoolDonateTest.sol b/src/test/PoolDonateTest.sol
index 2bf4aefc..7ee393db 100644
--- a/src/test/PoolDonateTest.sol
+++ b/src/test/PoolDonateTest.sol
@@ -30,7 +30,8 @@ contract PoolDonateTest is PoolTestBase, Test {
         returns (BalanceDelta delta)
     {
         delta = abi.decode(
-            manager.lock(abi.encode(CallbackData(msg.sender, key, amount0, amount1, hookData))), (BalanceDelta)
+            manager.lock(address(this), abi.encode(CallbackData(msg.sender, key, amount0, amount1, hookData))),
+            (BalanceDelta)
         );
 
         uint256 ethBalance = address(this).balance;
@@ -39,7 +40,7 @@ contract PoolDonateTest is PoolTestBase, Test {
         }
     }
 
-    function lockAcquired(bytes calldata rawData) external returns (bytes memory) {
+    function lockAcquired(address, bytes calldata rawData) external returns (bytes memory) {
         require(msg.sender == address(manager));
 
         CallbackData memory data = abi.decode(rawData, (CallbackData));
