diff --git a/src/test/PoolTakeTest.sol b/src/test/PoolTakeTest.sol
index 5a1cfdf9..4f5e9e0c 100644
--- a/src/test/PoolTakeTest.sol
+++ b/src/test/PoolTakeTest.sol
@@ -38,13 +38,13 @@ contract PoolTakeTest is Test, PoolTestBase {
 
     function _testTake(Currency currency, address sender, uint256 amount) internal {
         (uint256 userBalBefore, uint256 pmBalBefore, uint256 reserveBefore, int256 deltaBefore) =
-            _fetchBalances(currency, sender, address(this));
+            _fetchBalances(currency, sender);
         assertEq(deltaBefore, 0);
 
         _take(currency, sender, -(amount.toInt128()), true);
 
         (uint256 userBalAfter, uint256 pmBalAfter, uint256 reserveAfter, int256 deltaAfter) =
-            _fetchBalances(currency, sender, address(this));
+            _fetchBalances(currency, sender);
         assertEq(deltaAfter, amount.toInt128());
 
         assertEq(userBalAfter - userBalBefore, amount);
