diff --git a/contracts/PoolManager.sol b/contracts/PoolManager.sol
index 629af04a..edd8da5b 100644
--- a/contracts/PoolManager.sol
+++ b/contracts/PoolManager.sol
@@ -107,8 +107,6 @@ contract PoolManager is IPoolManager, Owned, NoDelegateCall, ERC1155, IERC1155Re
                 revert Hooks.InvalidHookResponse();
             }
         }
-
-        emit Initialize(id, key.token0, key.token1, key.fee, key.tickSpacing, key.hooks);
     }
 
     /// @inheritdoc IPoolManager
@@ -231,8 +229,7 @@ contract PoolManager is IPoolManager, Owned, NoDelegateCall, ERC1155, IERC1155Re
             }
         }
 
-        bytes32 poolId = key.toId();
-        delta = pools[poolId].modifyPosition(
+        delta = _getPool(key).modifyPosition(
             Pool.ModifyPositionParams({
                 owner: msg.sender,
                 tickLower: params.tickLower,
@@ -249,8 +246,6 @@ contract PoolManager is IPoolManager, Owned, NoDelegateCall, ERC1155, IERC1155Re
                 revert Hooks.InvalidHookResponse();
             }
         }
-
-        emit ModifyPosition(poolId, msg.sender, params.tickLower, params.tickUpper, params.liquidityDelta);
     }
 
     /// @inheritdoc IPoolManager
@@ -268,9 +263,7 @@ contract PoolManager is IPoolManager, Owned, NoDelegateCall, ERC1155, IERC1155Re
         }
 
         uint256 feeForProtocol;
-        Pool.SwapState memory state;
-        bytes32 poolId = key.toId();
-        (delta, feeForProtocol, state) = pools[poolId].swap(
+        (delta, feeForProtocol) = _getPool(key).swap(
             Pool.SwapParams({
                 fee: key.fee,
                 tickSpacing: key.tickSpacing,
@@ -292,8 +285,6 @@ contract PoolManager is IPoolManager, Owned, NoDelegateCall, ERC1155, IERC1155Re
                 revert Hooks.InvalidHookResponse();
             }
         }
-
-        emit Swap(poolId, msg.sender, delta.amount0, delta.amount1, state.sqrtPriceX96, state.liquidity, state.tick);
     }
 
     /// @inheritdoc IPoolManager
