diff --git a/test/foundry-tests/PoolManager.t.sol b/test/foundry-tests/PoolManager.t.sol
index 73ba98c2..f9b56560 100644
--- a/test/foundry-tests/PoolManager.t.sol
+++ b/test/foundry-tests/PoolManager.t.sol
@@ -1154,7 +1154,7 @@ contract PoolManagerTest is Test, Deployers, TokenFixture, GasSnapshot, IERC1155
         lockTest.lock();
     }
 
-    uint256 constant POOL_SLOT = 8;
+    uint256 constant POOL_SLOT = 4;
     uint256 constant TICKS_OFFSET = 4;
 
     function testExtsloadForPoolPrice() public {
