diff --git a/src/libraries/Pool.sol b/src/libraries/Pool.sol
index fed1aec6..71eb5a8e 100644
--- a/src/libraries/Pool.sol
+++ b/src/libraries/Pool.sol
@@ -402,10 +402,7 @@ library Pool {
                 }
             }
 
-            // Shift tick if we reached the next price, and preemptively decrement for zeroForOne swaps to tickNext - 1.
-            // If the swap doesnt continue (if amountRemaining == 0 or sqrtPriceLimit is met), slot0.tick will be 1 less
-            // than getTickAtSqrtPrice(slot0.sqrtPrice). This doesn't affect swaps, but donation calls should verify both
-            // price and tick to reward the correct LPs.
+            // shift tick if we reached the next price
             if (result.sqrtPriceX96 == step.sqrtPriceNextX96) {
                 // if the tick is initialized, run the tick transition
                 if (step.initialized) {
