diff --git a/src/PoolManager.sol b/src/PoolManager.sol
index 2b2e7f0b..18689f5f 100644
--- a/src/PoolManager.sol
+++ b/src/PoolManager.sol
@@ -178,13 +178,13 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         PoolKey memory key,
         IPoolManager.ModifyLiquidityParams memory params,
         bytes calldata hookData
-    ) external override onlyWhenUnlocked returns (BalanceDelta delta, BalanceDelta feeDelta) {
+    ) external override onlyWhenUnlocked returns (BalanceDelta delta) {
         PoolId id = key.toId();
         _checkPoolInitialized(id);
 
         key.hooks.beforeModifyLiquidity(key, params, hookData);
 
-        (delta, feeDelta) = pools[id].modifyLiquidity(
+        delta = pools[id].modifyLiquidity(
             Pool.ModifyLiquidityParams({
                 owner: msg.sender,
                 tickLower: params.tickLower,
@@ -194,7 +194,7 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
             })
         );
 
-        _accountPoolBalanceDelta(key, delta + feeDelta);
+        _accountPoolBalanceDelta(key, delta);
 
         emit ModifyLiquidity(id, msg.sender, params.tickLower, params.tickUpper, params.liquidityDelta);
 
