diff --git a/test/PoolManager.t.sol b/test/PoolManager.t.sol
index 8173b665..188389c6 100644
--- a/test/PoolManager.t.sol
+++ b/test/PoolManager.t.sol
@@ -35,7 +35,6 @@ import {SafeCast} from "../src/libraries/SafeCast.sol";
 import {AmountHelpers} from "./utils/AmountHelpers.sol";
 import {ProtocolFeeLibrary} from "../src/libraries/ProtocolFeeLibrary.sol";
 import {IProtocolFees} from "../src/interfaces/IProtocolFees.sol";
-import {StateLibrary} from "../src/libraries/StateLibrary.sol";
 
 contract PoolManagerTest is Test, Deployers, GasSnapshot {
     using Hooks for IHooks;
@@ -45,7 +44,6 @@ contract PoolManagerTest is Test, Deployers, GasSnapshot {
     using SafeCast for uint256;
     using SafeCast for uint128;
     using ProtocolFeeLibrary for uint24;
-    using StateLibrary for IPoolManager;
 
     event UnlockCallback();
     event ProtocolFeeControllerUpdated(address feeController);
