diff --git a/src/libraries/Pool.sol b/src/libraries/Pool.sol
index e7b6420c..271ba26f 100644
--- a/src/libraries/Pool.sol
+++ b/src/libraries/Pool.sol
@@ -472,15 +472,14 @@ library Pool {
 
     /// @notice Donates the given amount of currency0 and currency1 to the pool
     function donate(State storage state, uint256 amount0, uint256 amount1) internal returns (BalanceDelta delta) {
-        uint128 liquidity = state.liquidity;
-        if (liquidity == 0) revert NoLiquidityToReceiveFees();
+        if (state.liquidity == 0) revert NoLiquidityToReceiveFees();
         delta = toBalanceDelta(-(amount0.toInt128()), -(amount1.toInt128()));
         unchecked {
             if (amount0 > 0) {
-                state.feeGrowthGlobal0X128 += FullMath.mulDiv(amount0, FixedPoint128.Q128, liquidity);
+                state.feeGrowthGlobal0X128 += FullMath.mulDiv(amount0, FixedPoint128.Q128, state.liquidity);
             }
             if (amount1 > 0) {
-                state.feeGrowthGlobal1X128 += FullMath.mulDiv(amount1, FixedPoint128.Q128, liquidity);
+                state.feeGrowthGlobal1X128 += FullMath.mulDiv(amount1, FixedPoint128.Q128, state.liquidity);
             }
         }
     }
