diff --git a/contracts/PoolManager.sol b/contracts/PoolManager.sol
index 466979d1..1288a5e4 100644
--- a/contracts/PoolManager.sol
+++ b/contracts/PoolManager.sol
@@ -194,6 +194,7 @@ contract PoolManager is IPoolManager, NoDelegateCall, ERC1155, IERC1155Receiver
                 tickLower: params.tickLower,
                 tickUpper: params.tickUpper,
                 liquidityDelta: params.liquidityDelta.toInt128(),
+                maxLiquidityPerTick: Tick.tickSpacingToMaxLiquidityPerTick(key.tickSpacing),
                 tickSpacing: key.tickSpacing
             })
         );
