diff --git a/contracts/PoolManager.sol b/contracts/PoolManager.sol
index 216f8bdb..5045a445 100644
--- a/contracts/PoolManager.sol
+++ b/contracts/PoolManager.sol
@@ -56,7 +56,7 @@ contract PoolManager is IPoolManager, NoDelegateCall {
     mapping(IERC20Minimal => PositionAndDelta) public override tokenDelta;
 
     function lock(bytes calldata data) external override returns (bytes memory result) {
-        if (lockedBy != address(0)) revert AlreadyLocked(lockedBy);
+        require(lockedBy == address(0));
         lockedBy = msg.sender;
 
         // the caller does everything in this callback, including paying what they owe via calls to settle
@@ -85,7 +85,7 @@ contract PoolManager is IPoolManager, NoDelegateCall {
         slot = pd.slot;
 
         if (slot == 0 && tokensTouched[slot] != token) {
-            if (len >= type(uint8).max) revert MaxTokensTouched(token);
+            require(len < type(uint8).max);
             slot = uint8(len);
             pd.slot = slot;
             tokensTouched.push(token);
