diff --git a/src/test/PoolSwapTest.sol b/src/test/PoolSwapTest.sol
index 07484aa0..f9f5c4d7 100644
--- a/src/test/PoolSwapTest.sol
+++ b/src/test/PoolSwapTest.sol
@@ -32,6 +32,7 @@ contract PoolSwapTest is PoolTestBase {
     struct TestSettings {
         bool takeClaims;
         bool settleUsingBurn;
+        bool currencyAlreadySent;
     }
 
     function swap(
@@ -99,10 +100,22 @@ contract PoolSwapTest is PoolTestBase {
         }
 
         if (deltaAfter0 < 0) {
-            data.key.currency0.settle(manager, data.sender, uint256(-deltaAfter0), data.testSettings.settleUsingBurn);
+            if (data.testSettings.currencyAlreadySent) {
+                manager.settle(data.key.currency0);
+            } else {
+                data.key.currency0.settle(
+                    manager, data.sender, uint256(-deltaAfter0), data.testSettings.settleUsingBurn
+                );
+            }
         }
         if (deltaAfter1 < 0) {
-            data.key.currency1.settle(manager, data.sender, uint256(-deltaAfter1), data.testSettings.settleUsingBurn);
+            if (data.testSettings.currencyAlreadySent) {
+                manager.settle(data.key.currency1);
+            } else {
+                data.key.currency1.settle(
+                    manager, data.sender, uint256(-deltaAfter1), data.testSettings.settleUsingBurn
+                );
+            }
         }
         if (deltaAfter0 > 0) {
             data.key.currency0.take(manager, data.sender, uint256(deltaAfter0), data.testSettings.takeClaims);
