diff --git a/src/test/PoolEmptyLockTest.sol b/src/test/PoolEmptyLockTest.sol
index 7d4c39e3..05b59e30 100644
--- a/src/test/PoolEmptyLockTest.sol
+++ b/src/test/PoolEmptyLockTest.sol
@@ -14,11 +14,11 @@ contract PoolEmptyLockTest is ILockCallback {
     }
 
     function lock() external {
-        manager.lock("");
+        manager.lock(address(this), "");
     }
 
     /// @notice Called by the pool manager on `msg.sender` when a lock is acquired
-    function lockAcquired(bytes calldata) external override returns (bytes memory) {
+    function lockAcquired(address, bytes calldata) external override returns (bytes memory) {
         emit LockAcquired();
         return "";
     }
