diff --git a/src/test/PoolModifyPositionTest.sol b/src/test/PoolModifyPositionTest.sol
index 0a585252..c3b7a236 100644
--- a/src/test/PoolModifyPositionTest.sol
+++ b/src/test/PoolModifyPositionTest.sol
@@ -53,12 +53,12 @@ contract PoolModifyPositionTest is Test, PoolTestBase {
         (,,, int256 delta1) = _fetchBalances(data.key.currency1, data.sender);
 
         // These assertions only apply in non lock-accessing pools.
-        if (!data.key.hooks.hasPermission(Hooks.ACCESS_LOCK_FLAG)) {
+        if (!data.key.hooks.hasPermissionToAccessLock()) {
             if (data.params.liquidityDelta > 0) {
-                assert(delta0 > 0 || delta1 > 0 || data.key.hooks.hasPermission(Hooks.NO_OP_FLAG));
+                assert(delta0 > 0 || delta1 > 0 || data.key.hooks.hasPermissionToNoOp());
                 assert(!(delta0 < 0 || delta1 < 0));
             } else {
-                assert(delta0 < 0 || delta1 < 0 || data.key.hooks.hasPermission(Hooks.NO_OP_FLAG));
+                assert(delta0 < 0 || delta1 < 0 || data.key.hooks.hasPermissionToNoOp());
                 assert(!(delta0 > 0 || delta1 > 0));
             }
         }
