diff --git a/src/PoolManager.sol b/src/PoolManager.sol
index ad6a293d..ca7d2a32 100644
--- a/src/PoolManager.sol
+++ b/src/PoolManager.sol
@@ -20,7 +20,7 @@ import {BalanceDelta, BalanceDeltaLibrary} from "./types/BalanceDelta.sol";
 import {BeforeSwapDelta} from "./types/BeforeSwapDelta.sol";
 import {Lock} from "./libraries/Lock.sol";
 import {CurrencyDelta} from "./libraries/CurrencyDelta.sol";
-import {NonzeroDeltaCount} from "./libraries/NonzeroDeltaCount.sol";
+import {NonZeroDeltaCount} from "./libraries/NonZeroDeltaCount.sol";
 import {CurrencyReserves} from "./libraries/CurrencyReserves.sol";
 import {Extsload} from "./Extsload.sol";
 import {Exttload} from "./Exttload.sol";
@@ -110,7 +110,7 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         // the caller does everything in this callback, including paying what they owe via calls to settle
         result = IUnlockCallback(msg.sender).unlockCallback(data);
 
-        if (NonzeroDeltaCount.read() != 0) CurrencyNotSettled.selector.revertWith();
+        if (NonZeroDeltaCount.read() != 0) CurrencyNotSettled.selector.revertWith();
         Lock.lock();
     }
 
@@ -334,7 +334,7 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         if (currency.isNative()) {
             paid = msg.value;
         } else {
-            if (msg.value > 0) NonzeroNativeValue.selector.revertWith();
+            if (msg.value > 0) NonZeroNativeValue.selector.revertWith();
             // Reserves are guaranteed to be set, because currency and reserves are always set together
             uint256 reservesBefore = CurrencyReserves.getSyncedReserves();
             uint256 reservesNow = currency.balanceOfSelf();
@@ -351,9 +351,9 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         (int256 previous, int256 next) = currency.applyDelta(target, delta);
 
         if (next == 0) {
-            NonzeroDeltaCount.decrement();
+            NonZeroDeltaCount.decrement();
         } else if (previous == 0) {
-            NonzeroDeltaCount.increment();
+            NonZeroDeltaCount.increment();
         }
     }
 
