diff --git a/contracts/libraries/Position.sol b/contracts/libraries/Position.sol
index 7224377f..9525c195 100644
--- a/contracts/libraries/Position.sol
+++ b/contracts/libraries/Position.sol
@@ -5,6 +5,7 @@ import './FeeMath.sol';
 import './FullMath.sol';
 import './FixedPoint128.sol';
 import './LiquidityMath.sol';
+import './SafeMath.sol';
 
 /// @title Position
 /// @notice Positions represent an owner address' liquidity between a lower and upper tick boundary
