diff --git a/contracts/libraries/PriceMath.sol b/contracts/libraries/PriceMath.sol
index e20b1363..42a912c5 100644
--- a/contracts/libraries/PriceMath.sol
+++ b/contracts/libraries/PriceMath.sol
@@ -37,19 +37,7 @@ library PriceMath {
         // TODO this could probably be a bit safer/more elegant
         uint256 inputToRatio = getInputToRatioUQ144x112(reserveIn, reserveOut, lpFee, inOutRatio._x) + ROUND_UP;
         require(inputToRatio >> 112 <= uint112(-1), 'PriceMath: TODO');
-
-        amountIn = uint112(inputToRatio >> 112);
-
-        uint256 amountOut = ((uint256(reserveOut) * amountIn * (LP_FEE_BASE - lpFee)) /
-            (uint256(amountIn) * (LP_FEE_BASE - lpFee) + uint256(reserveIn) * LP_FEE_BASE));
-        uint256 reserveOutAfter = uint256(reserveOut) - amountOut;
-        uint256 reserveInAfter = uint256(reserveIn) + amountIn;
-        uint256 minReserveIn = FullMath.mulDiv(reserveOutAfter, inOutRatio._x, uint256(1) << 112, true);
-
-        if (minReserveIn > reserveInAfter) {
-            require(minReserveIn - reserveIn <= uint112(-1), 'PriceMath::getInputToRatio: amountIn overflows');
-            amountIn = uint112(minReserveIn - reserveIn);
-        }
+        return uint112(inputToRatio >> 112);
     }
 
     /**
