diff --git a/contracts/test/PriceMathEchidnaTest.sol b/contracts/test/PriceMathEchidnaTest.sol
index 9c76cda7..ea4f6047 100644
--- a/contracts/test/PriceMathEchidnaTest.sol
+++ b/contracts/test/PriceMathEchidnaTest.sol
@@ -19,7 +19,7 @@ contract PriceMathEchidnaTest {
         uint16 lpFee_,
         uint224 inOutRatio_
     ) external {
-        require(reserveIn > 101 && reserveOut > 101 && lpFee < PriceMath.LP_FEE_BASE);
+        require(reserveIn > 101 && reserveOut > 101);
 
         reserveIn = reserveIn_;
         reserveOut = reserveOut_;
@@ -33,8 +33,7 @@ contract PriceMathEchidnaTest {
         if (reserveIn == 0 || reserveOut == 0 || inOutRatio == 0) {
             return true;
         }
-        uint256 amountInLessFee = (uint256(amountIn) * (PriceMath.LP_FEE_BASE - lpFee)) / PriceMath.LP_FEE_BASE;
-        uint256 amountOut = reserveOut - ((uint256(reserveIn) * reserveOut) / (uint256(reserveIn) + amountInLessFee));
+        uint256 amountOut = (uint256(amountIn) * (PriceMath.LP_FEE_BASE - lpFee)) / PriceMath.LP_FEE_BASE;
         return ((uint256(reserveIn) + amountIn) << 112) / (uint256(reserveOut) - amountOut) >= inOutRatio;
     }
 }
