diff --git a/contracts/test/PriceMathEchidnaTest.sol b/contracts/test/PriceMathEchidnaTest.sol
index 88bd16a0..5db85053 100644
--- a/contracts/test/PriceMathEchidnaTest.sol
+++ b/contracts/test/PriceMathEchidnaTest.sol
@@ -28,9 +28,6 @@ contract PriceMathEchidnaTest {
     }
 
     function echidna_ratioAfterAmountInAlwaysExceedsPrice() external view returns (bool) {
-        if (reserveIn == 0 || reserveOut == 0 || lpFee == 0 || inOutRatio == 0) {
-            return true;
-        }
         uint112 amountOut = (amountIn * (PriceMath.LP_FEE_BASE - lpFee)) / PriceMath.LP_FEE_BASE;
         return ((reserveIn + amountIn) << 112) / (reserveOut - amountOut) >= inOutRatio;
     }
