diff --git a/src/test/ProxyPoolManager.sol b/src/test/ProxyPoolManager.sol
index b82bf842..f8f177be 100644
--- a/src/test/ProxyPoolManager.sol
+++ b/src/test/ProxyPoolManager.sol
@@ -141,14 +141,10 @@ contract ProxyPoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909
 
     /// @inheritdoc IPoolManager
     function sync(Currency currency) public {
-        // address(0) is used for the native currency
-        if (currency.isAddressZero()) {
-            // The reserves balance is not used for native settling, so we only need to reset the currency.
-            CurrencyReserves.resetCurrency();
-        } else {
-            uint256 balance = currency.balanceOfSelf();
-            CurrencyReserves.syncCurrencyAndReserves(currency, balance);
-        }
+        CurrencyReserves.requireNotSynced();
+        if (currency.isAddressZero()) return;
+        uint256 balance = currency.balanceOfSelf();
+        CurrencyReserves.syncCurrencyAndReserves(currency, balance);
     }
 
     /// @inheritdoc IPoolManager
