diff --git a/src/test/ProxyPoolManager.sol b/src/test/ProxyPoolManager.sol
index 496f5ced..ae278f98 100644
--- a/src/test/ProxyPoolManager.sol
+++ b/src/test/ProxyPoolManager.sol
@@ -15,7 +15,7 @@ import {IPoolManager} from "../interfaces/IPoolManager.sol";
 import {IUnlockCallback} from "../interfaces/callback/IUnlockCallback.sol";
 import {ProtocolFees} from "../ProtocolFees.sol";
 import {ERC6909Claims} from "../ERC6909Claims.sol";
-import {PoolId} from "../types/PoolId.sol";
+import {PoolId, PoolIdLibrary} from "../types/PoolId.sol";
 import {BalanceDelta} from "../types/BalanceDelta.sol";
 import {Lock} from "../libraries/Lock.sol";
 import {CurrencyDelta} from "../libraries/CurrencyDelta.sol";
@@ -27,6 +27,7 @@ import {CustomRevert} from "../libraries/CustomRevert.sol";
 
 /// @notice A proxy pool manager that delegates calls to the real/delegate pool manager
 contract ProxyPoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claims, Extsload, Exttload {
+    using PoolIdLibrary for PoolKey;
     using SafeCast for *;
     using Pool for *;
     using Hooks for IHooks;
