diff --git a/test/SkipCallsTestHook.t.sol b/test/SkipCallsTestHook.t.sol
index f95e7681..63fa1724 100644
--- a/test/SkipCallsTestHook.t.sol
+++ b/test/SkipCallsTestHook.t.sol
@@ -25,7 +25,8 @@ contract SkipCallsTest is Test, Deployers, GasSnapshot {
     IPoolManager.SwapParams swapParams =
         IPoolManager.SwapParams({zeroForOne: true, amountSpecified: -100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
 
-    PoolSwapTest.TestSettings testSettings = PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+    PoolSwapTest.TestSettings testSettings =
+        PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false, currencyAlreadySent: false});
 
     uint160 clearAllHookPermisssionsMask;
     uint256 hookPermissionCount = 10;
