diff --git a/contracts/libraries/SqrtPriceMath.sol b/contracts/libraries/SqrtPriceMath.sol
index 5108af12..32f71ef4 100644
--- a/contracts/libraries/SqrtPriceMath.sol
+++ b/contracts/libraries/SqrtPriceMath.sol
@@ -3,13 +3,13 @@ pragma solidity >=0.5.0;
 
 import './FeeMath.sol';
 import './FullMath.sol';
-import './CheapSafeMath.sol';
+import './SafeMath.sol';
 
 import './SafeCast.sol';
 import './FixedPoint96.sol';
 
 library SqrtPriceMath {
-    using CheapSafeMath for uint256;
+    using SafeMath for uint256;
     using SafeCast for uint256;
 
     // calculate liquidity * sqrt(P) / (liquidity +- x * sqrt(P))
