diff --git a/contracts/test/SqrtTickMathEchidnaTest.sol b/contracts/test/SqrtTickMathEchidnaTest.sol
index 8d472b50..c9457f08 100644
--- a/contracts/test/SqrtTickMathEchidnaTest.sol
+++ b/contracts/test/SqrtTickMathEchidnaTest.sol
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier: UNLICENSED
+// SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity =0.7.6;
 
 import '../libraries/SqrtTickMath.sol';
