diff --git a/contracts/libraries/SqrtTickMath.sol b/contracts/libraries/SqrtTickMath.sol
index fe48648f..f4c0d7f2 100644
--- a/contracts/libraries/SqrtTickMath.sol
+++ b/contracts/libraries/SqrtTickMath.sol
@@ -2,7 +2,7 @@
 pragma solidity >=0.5.0;
 
 import './TickMath.sol';
-import './UnsafeMath.sol';
+import './LowGasSafeMath.sol';
 
 /// @title Math library for computing sqrt(price) from ticks
 /// @notice Computes sqrt(price) from ticks of size 1.0001 as fixed point Q64.96 numbers
@@ -21,7 +21,7 @@ library SqrtTickMath {
     /// @param tick the tick for which to compute the sqrt price
     /// @return the sqrt price for ticks of size 1.0001
     function getSqrtRatioAtTick(int24 tick) internal pure returns (uint160) {
-        return uint160(UnsafeMath.divRoundingUp(TickMath.getRatioAtTick(tick), 1 << 32));
+        return uint160(LowGasSafeMath.divRoundingUp(TickMath.getRatioAtTick(tick), 1 << 32));
     }
 
     /// @notice Gets the tick from the sqrt(price)
