diff --git a/contracts/libraries/SqrtPriceMath.sol b/contracts/libraries/SqrtPriceMath.sol
index 11f02dca..d7f43fd2 100644
--- a/contracts/libraries/SqrtPriceMath.sol
+++ b/contracts/libraries/SqrtPriceMath.sol
@@ -12,7 +12,7 @@ library SqrtPriceMath {
     using SafeCast for uint256;
 
     function divRoundingUp(uint256 x, uint256 d) internal pure returns (uint256) {
-        // addition is safe because (uint256(-1) / 1) + (uint256(-1) % 1 > 0 ? 1 : 0) == uint256(-1)
+        // addition is safe because (type(uint256).max / 1) + (type(uint256).max % 1 > 0 ? 1 : 0) == type(uint256).max
         return (x / d) + (x % d > 0 ? 1 : 0);
     }
 
@@ -53,12 +53,12 @@ library SqrtPriceMath {
         uint256 quotient =
             add
                 ? (
-                    amount <= uint160(-1)
+                    amount <= type(uint160).max
                         ? (amount << FixedPoint96.RESOLUTION) / liquidity
                         : FullMath.mulDiv(amount, FixedPoint96.Q96, liquidity)
                 )
                 : (
-                    amount <= uint160(-1)
+                    amount <= type(uint160).max
                         ? divRoundingUp(amount << FixedPoint96.RESOLUTION, liquidity)
                         : FullMath.mulDivRoundingUp(amount, FixedPoint96.Q96, liquidity)
                 );
