diff --git a/contracts/test/SqrtPriceMathEchidnaTest.sol b/contracts/test/SqrtPriceMathEchidnaTest.sol
index c645491a..1e569acb 100644
--- a/contracts/test/SqrtPriceMathEchidnaTest.sol
+++ b/contracts/test/SqrtPriceMathEchidnaTest.sol
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: UNLICENSED
-pragma solidity =0.8.9;
+pragma solidity =0.7.6;
 
 import '../libraries/FullMath.sol';
 import '../libraries/SqrtPriceMath.sol';
@@ -11,17 +11,15 @@ contract SqrtPriceMathEchidnaTest {
         uint256 y,
         uint256 z
     ) external pure {
-        unchecked {
-            require(z > 0);
-            uint256 notRoundedUp = FullMath.mulDiv(x, y, z);
-            uint256 roundedUp = FullMath.mulDivRoundingUp(x, y, z);
-            assert(roundedUp >= notRoundedUp);
-            assert(roundedUp - notRoundedUp < 2);
-            if (roundedUp - notRoundedUp == 1) {
-                assert(mulmod(x, y, z) > 0);
-            } else {
-                assert(mulmod(x, y, z) == 0);
-            }
+        require(z > 0);
+        uint256 notRoundedUp = FullMath.mulDiv(x, y, z);
+        uint256 roundedUp = FullMath.mulDivRoundingUp(x, y, z);
+        assert(roundedUp >= notRoundedUp);
+        assert(roundedUp - notRoundedUp < 2);
+        if (roundedUp - notRoundedUp == 1) {
+            assert(mulmod(x, y, z) > 0);
+        } else {
+            assert(mulmod(x, y, z) == 0);
         }
     }
 
