diff --git a/contracts/test/SqrtPriceMathTest.sol b/contracts/test/SqrtPriceMathTest.sol
index 44fc453d..5a456dc8 100644
--- a/contracts/test/SqrtPriceMathTest.sol
+++ b/contracts/test/SqrtPriceMathTest.sol
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
-pragma solidity =0.7.6;
-pragma abicoder v2;
+pragma solidity =0.6.12;
+pragma experimental ABIEncoderV2;
 
 import '../libraries/SqrtPriceMath.sol';
 import '../libraries/FixedPoint96.sol';
