diff --git a/src/libraries/StateLibrary.sol b/src/libraries/StateLibrary.sol
index ec3d9f3c..397ad276 100644
--- a/src/libraries/StateLibrary.sol
+++ b/src/libraries/StateLibrary.sol
@@ -52,7 +52,7 @@ library StateLibrary {
         //   24 bits  |24bits|24bits      |24 bits|160 bits
         // 0x000000   |000bb8|000000      |ffff75 |0000000000000000fe3aa841ba359daa0ea9eff7
         // ---------- | fee  |protocolfee | tick  | sqrtPriceX96
-        assembly ("memory-safe") {
+        assembly {
             // bottom 160 bits of data
             sqrtPriceX96 := and(data, 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)
             // next 24 bits of data
@@ -89,7 +89,7 @@ library StateLibrary {
 
         // read all 3 words of the TickInfo struct
         bytes memory data = manager.extsload(slot, 3);
-        assembly ("memory-safe") {
+        assembly {
             let firstWord := mload(add(data, 32))
             liquidityNet := sar(128, firstWord)
             liquidityGross := and(firstWord, 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)
@@ -115,7 +115,7 @@ library StateLibrary {
         bytes32 slot = _getTickInfoSlot(poolId, tick);
 
         bytes32 value = manager.extsload(slot);
-        assembly ("memory-safe") {
+        assembly {
             liquidityNet := sar(128, value)
             liquidityGross := and(value, 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)
         }
@@ -139,7 +139,7 @@ library StateLibrary {
 
         // offset by 1 word, since the first word is liquidityGross + liquidityNet
         bytes memory data = manager.extsload(bytes32(uint256(slot) + 1), 2);
-        assembly ("memory-safe") {
+        assembly {
             feeGrowthOutside0X128 := mload(add(data, 32))
             feeGrowthOutside1X128 := mload(add(data, 64))
         }
@@ -166,7 +166,7 @@ library StateLibrary {
 
         // read the 2 words of feeGrowthGlobal
         bytes memory data = manager.extsload(slot_feeGrowthGlobal0X128, 2);
-        assembly ("memory-safe") {
+        assembly {
             feeGrowthGlobal0 := mload(add(data, 32))
             feeGrowthGlobal1 := mload(add(data, 64))
         }
@@ -234,7 +234,7 @@ library StateLibrary {
         // read all 3 words of the Position.Info struct
         bytes memory data = manager.extsload(slot, 3);
 
-        assembly ("memory-safe") {
+        assembly {
             liquidity := mload(add(data, 32))
             feeGrowthInside0LastX128 := mload(add(data, 64))
             feeGrowthInside1LastX128 := mload(add(data, 96))
