diff --git a/contracts/test/SwapMathTest.sol b/contracts/test/SwapMathTest.sol
index 714894ad..07056f97 100644
--- a/contracts/test/SwapMathTest.sol
+++ b/contracts/test/SwapMathTest.sol
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity =0.8.19;
 
-import {SwapMath} from "../libraries/SwapMath.sol";
+import {SwapMath} from '../libraries/SwapMath.sol';
 
 contract SwapMathTest {
     function computeSwapStep(
