diff --git a/test/utils/SwapHelper.t.sol b/test/utils/SwapHelper.t.sol
index 075e1169..9edd8c07 100644
--- a/test/utils/SwapHelper.t.sol
+++ b/test/utils/SwapHelper.t.sol
@@ -19,19 +19,19 @@ import {PoolId, PoolIdLibrary} from "../../src/types/PoolId.sol";
 import {PoolKey} from "../../src/types/PoolKey.sol";
 import {IERC20Minimal} from "../../src/interfaces/external/IERC20Minimal.sol";
 import {BalanceDelta} from "../../src/types/BalanceDelta.sol";
-import {Constants} from "../utils/Constants.sol";
 
 /// @notice Testing Deployers.swap() and Deployers.swapNativeInput()
 contract SwapHelperTest is Test, Deployers, GasSnapshot {
     using PoolIdLibrary for PoolKey;
     using Hooks for IHooks;
 
+    address payable ALL_HOOKS_ADDRESS = payable(0x0000000000000000000000000000000000003fFF);
     MockHooks mockHooks;
 
     function setUp() public {
         MockHooks impl = new MockHooks();
-        vm.etch(Constants.ALL_HOOKS, address(impl).code);
-        mockHooks = MockHooks(Constants.ALL_HOOKS);
+        vm.etch(ALL_HOOKS_ADDRESS, address(impl).code);
+        mockHooks = MockHooks(ALL_HOOKS_ADDRESS);
 
         initializeManagerRoutersAndPoolsWithLiq(mockHooks);
     }
