diff --git a/contracts/test/SwapMathTest.sol b/contracts/test/SwapMathTest.sol
index 156498e3..4496efc9 100644
--- a/contracts/test/SwapMathTest.sol
+++ b/contracts/test/SwapMathTest.sol
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
-pragma solidity =0.7.6;
-pragma abicoder v2;
+pragma solidity =0.6.12;
+pragma experimental ABIEncoderV2;
 
 import '../libraries/SwapMath.sol';
 import '../libraries/FixedPoint96.sol';
