diff --git a/contracts/test/SwapMathEchidnaTest.sol b/contracts/test/SwapMathEchidnaTest.sol
index c03277bc..67bf8612 100644
--- a/contracts/test/SwapMathEchidnaTest.sol
+++ b/contracts/test/SwapMathEchidnaTest.sol
@@ -19,7 +19,7 @@ contract SwapMathEchidnaTest {
         (uint160 sqrtQ, uint256 amountIn, uint256 amountOut, uint256 feeAmount) =
             SwapMath.computeSwapStep(sqrtPriceRaw, sqrtPriceTargetRaw, liquidity, amountRemaining, feePips);
 
-        assert(amountIn <= uint256(-1) - feeAmount);
+        assert(amountIn <= type(uint256).max - feeAmount);
 
         if (amountRemaining < 0) {
             assert(amountOut <= uint256(-amountRemaining));
