diff --git a/contracts/test/TickOverflowSafetyEchidnaTest.sol b/contracts/test/TickOverflowSafetyEchidnaTest.sol
index a76f05aa..2613be34 100644
--- a/contracts/test/TickOverflowSafetyEchidnaTest.sol
+++ b/contracts/test/TickOverflowSafetyEchidnaTest.sol
@@ -8,7 +8,7 @@ contract TickOverflowSafetyEchidnaTest {
 
     int24 private constant MIN_TICK = -16;
     int24 private constant MAX_TICK = 16;
-    uint128 private constant MAX_LIQUIDITY = uint128(-1) / 32;
+    uint128 private constant MAX_LIQUIDITY = type(uint128).max / 32;
 
     mapping(int24 => Tick.Info) private ticks;
     int24 private tick = 0;
@@ -16,8 +16,8 @@ contract TickOverflowSafetyEchidnaTest {
     // used to track how much total liquidity has been added. should never be negative
     int256 totalLiquidity = 0;
     // half the cap of fee growth has happened, this can overflow
-    uint256 private feeGrowthGlobal0X128 = uint256(-1) / 2;
-    uint256 private feeGrowthGlobal1X128 = uint256(-1) / 2;
+    uint256 private feeGrowthGlobal0X128 = type(uint256).max / 2;
+    uint256 private feeGrowthGlobal1X128 = type(uint256).max / 2;
     // how much total growth has happened, this cannot overflow
     uint256 private totalGrowth0 = 0;
     uint256 private totalGrowth1 = 0;
