diff --git a/src/libraries/TickBitmap.sol b/src/libraries/TickBitmap.sol
index dc9547d1..2ffbfbbf 100644
--- a/src/libraries/TickBitmap.sol
+++ b/src/libraries/TickBitmap.sol
@@ -43,11 +43,13 @@ library TickBitmap {
     /// @param tick The tick to flip
     /// @param tickSpacing The spacing between usable ticks
     function flipTick(mapping(int16 => uint256) storage self, int24 tick, int24 tickSpacing) internal {
-        // Equivalent to the following Solidity:
-        //     if (tick % tickSpacing != 0) revert TickMisaligned(tick, tickSpacing);
-        //     (int16 wordPos, uint8 bitPos) = position(tick / tickSpacing);
-        //     uint256 mask = 1 << bitPos;
-        //     self[wordPos] ^= mask;
+        /*
+         * Equivalent to the following Solidity:
+         *     if (tick % tickSpacing != 0) revert TickMisaligned(tick, tickSpacing);
+         *     (int16 wordPos, uint8 bitPos) = position(tick / tickSpacing);
+         *     uint256 mask = 1 << bitPos;
+         *     self[wordPos] ^= mask;
+         */
         assembly ("memory-safe") {
             // ensure that the tick is spaced
             if smod(tick, tickSpacing) {
