diff --git a/contracts/test/TickMathTest.sol b/contracts/test/TickMathTest.sol
index 0f993045..48fe02db 100644
--- a/contracts/test/TickMathTest.sol
+++ b/contracts/test/TickMathTest.sol
@@ -2,11 +2,12 @@
 pragma solidity =0.7.6;
 pragma abicoder v2;
 
+import '../libraries/FixedPoint128.sol';
 import '../libraries/TickMath.sol';
 
 contract TickMathTest {
-    function getRatioAtTick(int24 tick) public pure returns (uint256) {
-        return TickMath.getRatioAtTick(tick);
+    function getRatioAtTick(int24 tick) public pure returns (FixedPoint128.uq128x128 memory) {
+        return FixedPoint128.uq128x128(TickMath.getRatioAtTick(tick));
     }
 
     function getRatioAtTickGasUsed(int24 tick) public view returns (uint256) {
@@ -16,13 +17,13 @@ contract TickMathTest {
         return (gasBefore - gasAfter);
     }
 
-    function getTickAtRatio(uint256 price) public pure returns (int24 tick) {
-        return TickMath.getTickAtRatio(price);
+    function getTickAtRatio(FixedPoint128.uq128x128 memory price) public pure returns (int24 tick) {
+        return TickMath.getTickAtRatio(price._x);
     }
 
-    function getTickAtRatioGasUsed(uint256 price) public view returns (uint256) {
+    function getTickAtRatioGasUsed(FixedPoint128.uq128x128 memory price) public view returns (uint256) {
         uint256 gasBefore = gasleft();
-        TickMath.getTickAtRatio(price);
+        TickMath.getTickAtRatio(price._x);
         uint256 gasAfter = gasleft();
         return (gasBefore - gasAfter);
     }
