diff --git a/src/libraries/TickMath.sol b/src/libraries/TickMath.sol
index 3b0c73e7..a0c0f218 100644
--- a/src/libraries/TickMath.sol
+++ b/src/libraries/TickMath.sol
@@ -51,11 +51,11 @@ library TickMath {
         unchecked {
             uint256 absTick;
             assembly {
-                // mask = 0 if tick >= 0 else -1 (all 1s)
+                // mask = 0 if tick >= 0 else -1
                 let mask := sar(255, tick)
-                // if tick >= 0, |tick| = tick = 0 ^ tick
-                // if tick < 0, |tick| = ~~|tick| = ~(-|tick| - 1) = ~(tick - 1) = (-1) ^ (tick - 1)
-                // either way, |tick| = mask ^ (tick + mask)
+                // If tick >= 0, |tick| = tick = 0 ^ tick
+                // If tick < 0, |tick| = ~~|tick| = ~(-|tick| - 1) = ~(tick - 1) = (-1) ^ (tick - 1)
+                // Either case, |tick| = mask ^ (tick + mask)
                 absTick := xor(mask, add(mask, tick))
             }
             // Equivalent: if (absTick > MAX_TICK) revert InvalidTick();
