diff --git a/src/libraries/TickMath.sol b/src/libraries/TickMath.sol
index 5a43775e..3b0c73e7 100644
--- a/src/libraries/TickMath.sol
+++ b/src/libraries/TickMath.sol
@@ -68,13 +68,8 @@ library TickMath {
                 }
             }
 
-            // Equivalent to:
-            //     price = absTick & 0x1 != 0 ? 0xfffcb933bd6fad37aa2d162d1a594001 : 0x100000000000000000000000000000000;
-            //     or price = int(2**128 / sqrt(1.0001)) if (absTick & 0x1) else 1 << 128
-            uint256 price;
-            assembly {
-                price := xor(shl(128, 1), mul(xor(shl(128, 1), 0xfffcb933bd6fad37aa2d162d1a594001), and(absTick, 0x1)))
-            }
+            uint256 price =
+                absTick & 0x1 != 0 ? 0xfffcb933bd6fad37aa2d162d1a594001 : 0x100000000000000000000000000000000;
             if (absTick & 0x2 != 0) price = (price * 0xfff97272373d413259a46990580e213a) >> 128;
             if (absTick & 0x4 != 0) price = (price * 0xfff2e50f5f656932ef12357cf3c7fdcc) >> 128;
             if (absTick & 0x8 != 0) price = (price * 0xffe5caca7e10e4e61c3624eaa0941cd0) >> 128;
