diff --git a/contracts/test/TickMathEchidnaTest.sol b/contracts/test/TickMathEchidnaTest.sol
index 79c2ac07..f5c68674 100644
--- a/contracts/test/TickMathEchidnaTest.sol
+++ b/contracts/test/TickMathEchidnaTest.sol
@@ -5,13 +5,13 @@ import '../libraries/TickMath.sol';
 
 contract TickMathEchidnaTest {
     // uniqueness and increasing order
-    function checkGetRatioAtTickInvariants(int24 tick) external pure {
+    function checkGetRatioAtTickInvariants(int24 tick) public pure {
         uint256 ratio = TickMath.getRatioAtTick(tick);
         assert(TickMath.getRatioAtTick(tick - 1) < ratio && ratio < TickMath.getRatioAtTick(tick + 1));
     }
 
     // the ratio is always between the returned tick and the returned tick+1
-    function checkGetTickAtRatioInvariants(uint256 ratio) external pure {
+    function checkGetTickAtRatioInvariants(uint256 ratio) public pure {
         int24 tick = TickMath.getTickAtRatio(ratio);
         assert(ratio >= TickMath.getRatioAtTick(tick) && ratio < TickMath.getRatioAtTick(tick + 1));
     }
