diff --git a/contracts/libraries/Tick.sol b/contracts/libraries/Tick.sol
index fa7ccc58..522aefe3 100644
--- a/contracts/libraries/Tick.sol
+++ b/contracts/libraries/Tick.sol
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity >=0.5.0;
 
-import './TickMath.sol';
+import './SqrtTickMath.sol';
 import './SafeCast.sol';
 import './LiquidityMath.sol';
 import './LowGasSafeMath.sol';
@@ -27,8 +27,8 @@ library Tick {
     ///     e.g., a tickSpacing of 3 requires ticks to be initialized every 3rd tick i.e., ..., -6, -3, 0, 3, 6, ...
     /// @return The max liquidity per tick
     function tickSpacingToMaxLiquidityPerTick(int24 tickSpacing) internal pure returns (uint128) {
-        int24 minTick = (TickMath.MIN_TICK / tickSpacing) * tickSpacing;
-        int24 maxTick = (TickMath.MAX_TICK / tickSpacing) * tickSpacing;
+        int24 minTick = (SqrtTickMath.MIN_TICK / tickSpacing) * tickSpacing;
+        int24 maxTick = (SqrtTickMath.MAX_TICK / tickSpacing) * tickSpacing;
         uint24 numTicks = uint24((maxTick - minTick) / tickSpacing) + 1;
         return type(uint128).max / numTicks;
     }
