diff --git a/contracts/test/TickBitmapTest.sol b/contracts/test/TickBitmapTest.sol
index f7b4b924..902561d2 100644
--- a/contracts/test/TickBitmapTest.sol
+++ b/contracts/test/TickBitmapTest.sol
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity =0.8.19;
 
-import {TickBitmap} from "../libraries/TickBitmap.sol";
+import {TickBitmap} from '../libraries/TickBitmap.sol';
 
 contract TickBitmapTest {
     using TickBitmap for mapping(int16 => uint256);
@@ -18,11 +18,10 @@ contract TickBitmapTest {
         return gasBefore - gasleft();
     }
 
-    function nextInitializedTickWithinOneWord(int24 tick, bool lte)
-        external
-        view
-        returns (int24 next, bool initialized)
-    {
+    function nextInitializedTickWithinOneWord(
+        int24 tick,
+        bool lte
+    ) external view returns (int24 next, bool initialized) {
         return bitmap.nextInitializedTickWithinOneWord(tick, 1, lte);
     }
 
