diff --git a/contracts/libraries/Tick.sol b/contracts/libraries/Tick.sol
index 98de0e6b..c7e54818 100644
--- a/contracts/libraries/Tick.sol
+++ b/contracts/libraries/Tick.sol
@@ -26,7 +26,7 @@ library Tick {
     }
 
     /// @notice Derives max liquidity per tick from given tick spacing
-    /// @dev Executed within the pool constructor
+    /// @dev Executed within the pair constructor
     /// @param tickSpacing The amount of required tick separation, realized in multiples of `tickSpacing`
     ///     e.g., a tickSpacing of 3 requires ticks to be initialized every 3rd tick i.e., ..., -6, -3, 0, 3, 6, ...
     /// @return The max liquidity per tick
