diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index 81738653..f6e5036a 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -30,7 +30,7 @@ contract UniswapV3Pair is IUniswapV3Pair, TickMath1r01 {
     using SafeCast for uint256;
     using MixedSafeMath for uint128;
     using FixedPoint128 for FixedPoint128.uq128x128;
-    using TickBitMap for mapping(int16 => uint256);
+    using TickBitMap for mapping(uint256 => uint256);
 
     // if we constrain the liquidity associated to a single tick, then we can guarantee that the total
     // liquidityCurrent never exceeds uint128
@@ -49,7 +49,7 @@ contract UniswapV3Pair is IUniswapV3Pair, TickMath1r01 {
     address public override feeTo;
 
     // see TickBitMap.sol
-    mapping(int16 => uint256) public override tickBitMap;
+    mapping(uint256 => uint256) public override tickBitMap;
 
     // single storage slot
     uint32 public override blockTimestampLast;
