diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index 5bfea06b..876786c1 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -228,19 +228,22 @@ contract UniswapV3Pair is IUniswapV3Pair {
         aggregateFeeVote = aggregateFeeVote.add(deltaFeeVote);
     }
 
-    function _getOrInitializeTick(int16 tickIndex, int16 _currentTick) internal {
-        if (tickInfos[tickIndex].kGrowthOutside._x == 0) {
+    function _getOrInitializeTick(int16 tickIndex, int16 _currentTick) internal returns (TickInfo memory tickInfo) {
+        tickInfo = tickInfos[tickIndex];
+        if (tickInfo.kGrowthOutside._x == 0) {
             // by convention, we assume that all growth before tick was initialized happened _below_ a tick
             if (tickIndex <= _currentTick) {
-                tickInfos[tickIndex] = TickInfo({
+                tickInfo = TickInfo({
                     secondsGrowthOutside: uint32(now % 2**32),
-                    kGrowthOutside: getInvariant()
+                    kGrowthOutside: getInvariant();
                 });
+                tickInfos[tickIndex] = tickInfo;
             } else {
                 tickInfos[tickIndex] = TickInfo({
                     secondsGrowthOutside: 0,
                     kGrowthOutside: FixedPoint.encode(1)
                 });
+                tickInfos[tickIndex] = tickInfo;
             }
         }
     }
