diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index a8702b38..01438cd4 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -280,7 +280,7 @@ contract UniswapV3Pair is IUniswapV3Pair {
 
         // if it's possible this position accrued any fees, rebate them
         if (position.liquidityAdjusted > 0) {
-            // TODO is this correct/precise?
+            // TODO is this precise/correct?
             int112 liquidityFee =
                 uint(FixedPoint.decode144(growthInside.mul(position.liquidityAdjusted)))
                 .sub(position.liquidity)
@@ -290,9 +290,11 @@ contract UniswapV3Pair is IUniswapV3Pair {
                 address feeTo = IUniswapV3Factory(factory).feeTo();
                 bool feeOn = feeTo != address(0);
                 if (feeOn && msg.sender != feeTo) {
-                    // TODO do the right thing here
                     int112 liquidityProtocol = liquidityFee / 6;
-                    liquidityFee -= liquidityProtocol;
+                    if (liquidityProtocol > 0) {
+                        liquidityFee -= liquidityProtocol;
+                        // TODO do the right thing here
+                    }
                 }
 
                 (amount0, amount1) = getValueAtPrice(price, -liquidityFee);
@@ -408,13 +410,18 @@ contract UniswapV3Pair is IUniswapV3Pair {
                     continue;
                 }
 
-                // calculate the amount of reserves + liquidity to kick in/out
+                // kick in/out reserves
                 int112 token0VirtualDelta =
                     tickInfo.token0VirtualDeltas[uint8(FeeVote.FeeVote0)] +
                     tickInfo.token0VirtualDeltas[uint8(FeeVote.FeeVote1)] +
                     tickInfo.token0VirtualDeltas[uint8(FeeVote.FeeVote2)] +
                     tickInfo.token0VirtualDeltas[uint8(FeeVote.FeeVote3)];
                 int112 token1VirtualDelta = FixedPointExtra.muli(price, token0VirtualDelta).itoInt112();
+                // subi because we're moving from right to left
+                reserve0Virtual = reserve0Virtual.subi(token0VirtualDelta).toUint112();
+                reserve1Virtual = reserve1Virtual.subi(token1VirtualDelta).toUint112();
+
+                // kick in/out liquidity
                 uint112 virtualSupply = getVirtualSupply();
                 int112[4] memory virtualSupplyDeltas = [
                     (tickInfo.token0VirtualDeltas[uint8(FeeVote.FeeVote0)].imul(virtualSupply) / reserve0Virtual)
@@ -426,10 +433,7 @@ contract UniswapV3Pair is IUniswapV3Pair {
                     (tickInfo.token0VirtualDeltas[uint8(FeeVote.FeeVote3)].imul(virtualSupply) / reserve0Virtual)
                         .itoInt112()
                 ];
-
                 // subi because we're moving from right to left
-                reserve0Virtual = reserve0Virtual.subi(token0VirtualDelta).toUint112();
-                reserve1Virtual = reserve1Virtual.subi(token1VirtualDelta).toUint112();
                 virtualSupplies = [
                     virtualSupplies[uint8(FeeVote.FeeVote0)].subi(virtualSupplyDeltas[uint8(FeeVote.FeeVote0)])
                         .toUint112(),
