diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index 12ab8aa3..728cc6a5 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -25,7 +25,6 @@ import './interfaces/IUniswapV3PairDeployer.sol';
 import './interfaces/IUniswapV3Factory.sol';
 import './interfaces/callback/IUniswapV3MintCallback.sol';
 import './interfaces/callback/IUniswapV3SwapCallback.sol';
-import './interfaces/callback/IUniswapV3FlashCallback.sol';
 
 contract UniswapV3Pair is IUniswapV3Pair {
     using SafeMath for uint256;
@@ -617,35 +616,4 @@ contract UniswapV3Pair is IUniswapV3Pair {
 
         emit CollectProtocol(recipient, amount0, amount1);
     }
-
-    function flash(
-        address recipient,
-        uint256 amount0,
-        uint256 amount1,
-        bytes calldata data
-    ) external override lock {
-        uint128 _liquidity = liquidity;
-        require(_liquidity > 0, 'L');
-
-        uint256 fee0 = SqrtPriceMath.mulDivRoundingUp(amount0, fee, 1e6);
-        uint256 fee1 = SqrtPriceMath.mulDivRoundingUp(amount1, fee, 1e6);
-        uint256 balance0Before = balance0();
-        uint256 balance1Before = balance1();
-
-        if (amount0 > 0) TransferHelper.safeTransfer(token0, recipient, amount0);
-        if (amount1 > 0) TransferHelper.safeTransfer(token1, recipient, amount1);
-
-        IUniswapV3FlashCallback(msg.sender).uniswapV3FlashCallback(fee0, fee1, data);
-
-        uint256 paid0 = balance0().sub(balance0Before);
-        uint256 paid1 = balance1().sub(balance1Before);
-
-        require(paid0 >= fee0, 'F0');
-        require(paid1 >= fee1, 'F1');
-
-        if (paid0 > 0) feeGrowthGlobal0X128 += FullMath.mulDiv(paid0, FixedPoint128.Q128, _liquidity);
-        if (paid1 > 0) feeGrowthGlobal1X128 += FullMath.mulDiv(paid1, FixedPoint128.Q128, _liquidity);
-
-        emit Flash(msg.sender, recipient, amount0, amount1, paid0, paid1);
-    }
 }
