diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index 1e3cd1f9..9b642498 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -6,7 +6,7 @@ import './libraries/FullMath.sol';
 import './libraries/TransferHelper.sol';
 
 import './libraries/FeeMath.sol';
-import './libraries/LowGasSafeMath.sol';
+import './libraries/CheapSafeMath.sol';
 
 import './libraries/SafeCast.sol';
 import './libraries/LiquidityMath.sol';
@@ -30,8 +30,8 @@ import './interfaces/callback/IUniswapV3FlashCallback.sol';
 import './NoDelegateCall.sol';
 
 contract UniswapV3Pair is IUniswapV3Pair, NoDelegateCall {
-    using LowGasSafeMath for uint256;
-    using LowGasSafeMath for int256;
+    using CheapSafeMath for uint256;
+    using CheapSafeMath for int256;
     using SafeCast for uint256;
     using SafeCast for int256;
     using LiquidityMath for uint128;
