diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index dcba75af..c57a9d0c 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -609,7 +609,12 @@ contract UniswapV3Pair is IUniswapV3Pair {
 
         Slot0 memory _slot0 = slot0;
         require(_slot0.unlockedAndPriceBit & UNLOCKED_BIT == UNLOCKED_BIT, 'LOK');
-        require(zeroForOne ? sqrtPriceLimitX96 < _slot0.sqrtPriceX96 : sqrtPriceLimitX96 > _slot0.sqrtPriceX96, 'SPL');
+        require(
+            zeroForOne
+                ? sqrtPriceLimitX96 < _slot0.sqrtPriceX96
+                : sqrtPriceLimitX96 > _slot0.sqrtPriceX96,
+            'SPL'
+        );
 
         _swap(
             SwapParams({
