diff --git a/contracts/UniswapV2Pair.sol b/contracts/UniswapV2Pair.sol
index f87a1db2..1fa9d3cb 100644
--- a/contracts/UniswapV2Pair.sol
+++ b/contracts/UniswapV2Pair.sol
@@ -13,7 +13,7 @@ contract UniswapV2Pair is IUniswapV2Pair, UniswapV2ERC20 {
     using UQ112x112 for uint224;
 
     uint public constant MINIMUM_LIQUIDITY = 10**3;
-    bytes4 private constant SELECTOR = bytes4(keccak256(bytes('transfer(address,uint256)')));
+    bytes4 private constant SELECTOR = 0xa9059cbb; // transfer(address,uint256)
 
     address public factory;
     address public token0;
@@ -59,6 +59,7 @@ contract UniswapV2Pair is IUniswapV2Pair, UniswapV2ERC20 {
     event Sync(uint112 reserve0, uint112 reserve1);
 
     constructor() public {
+        require(SELECTOR == IERC20(0).transfer.selector);
         factory = msg.sender;
     }
 
