diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index 0dbba0ee..3dcc2ea0 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -444,9 +444,7 @@ contract UniswapV3Pair is IUniswapV3Pair {
     }
 
     struct SwapParams {
-        // the liquidity at the beginning of the swap
-        uint128 liquidityStart;
-        // the tick at the beginning of the swap
+        // the tick where the price starts
         int24 tickStart;
         // whether the swap is from token 0 to 1, or 1 for 0
         bool zeroForOne;
@@ -503,7 +501,7 @@ contract UniswapV3Pair is IUniswapV3Pair {
                 tick: params.tickStart,
                 sqrtPrice: slot0.sqrtPriceCurrent,
                 feeGrowthGlobal: params.zeroForOne ? feeGrowthGlobal0 : feeGrowthGlobal1,
-                liquidityCurrent: params.liquidityStart
+                liquidityCurrent: slot1.liquidityCurrent
             });
 
         while (state.amountSpecifiedRemaining != 0) {
@@ -585,13 +583,10 @@ contract UniswapV3Pair is IUniswapV3Pair {
             }
         }
 
-        // update liquidity if it changed
-        if (params.liquidityStart != state.liquidityCurrent) {
+        // the price moved at least one tick
+        if (state.tick != params.tickStart) {
             slot1.liquidityCurrent = state.liquidityCurrent;
-        }
 
-        // the price moved at least one tick, update the accumulator
-        if (state.tick != params.tickStart) {
             uint32 _blockTimestampLast = slot0.blockTimestampLast;
             if (_blockTimestampLast != params.blockTimestamp) {
                 slot0.blockTimestampLast = params.blockTimestamp;
@@ -648,7 +643,6 @@ contract UniswapV3Pair is IUniswapV3Pair {
         return
             _swap(
                 SwapParams({
-                    liquidityStart: slot1.liquidityCurrent,
                     tickStart: tickCurrent(),
                     zeroForOne: zeroForOne,
                     amountSpecified: amountSpecified,
