diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index aac2a336..ed3fc981 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -228,14 +228,11 @@ contract UniswapV3Pair is IUniswapV3Pair {
             lastAdjustedLiquidity: liquidity,
             feeVote: feeVote
         });
-        uint112 totalLiquidity = liquidity + MINIMUM_LIQUIDITY;
-        virtualSupply = totalLiquidity;
+        virtualSupply = liquidity + MINIMUM_LIQUIDITY;
         
         // update fee
-        updateFee(Aggregate({
-            numerator: int112(feeVote * totalLiquidity),
-            denominator: int112(totalLiquidity)
-        }));
+        totalFeeVote = feeVote * liquidity;
+        lpFee = feeVote;
 
         uint112 _reserve0 = amount0;
         uint112 _reserve1 = amount1;
