diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index 97ecbd37..d6712013 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -564,6 +564,7 @@ contract UniswapV3Pair is IUniswapV3Pair {
                 Tick.Info storage tickInfo = tickInfos[step.tickNext];
 
                 // if the tick is initialized, update it
+                // todo: decide on a minimum here that may be non-zero
                 if (tickInfo.liquidityGross > 0) {
                     // update tick info
                     tickInfo.feeGrowthOutside0 = FixedPoint128.uq128x128(
@@ -584,9 +585,8 @@ contract UniswapV3Pair is IUniswapV3Pair {
                     }
                 }
 
-                // todo: this might not be ok. the price may not move due to the remaining input amount!
-                // original rationale: this is ok because we still have amountInRemaining so price is guaranteed to be
-                // less than the tick after swapping the remaining amount in
+                // this is ok because we still have amountInRemaining so price is guaranteed to be less than the tick
+                // after swapping the remaining amount in
                 state.tick = params.zeroForOne ? step.tickNext - 1 : step.tickNext;
             } else {
                 state.tick = SqrtTickMath.getTickAtSqrtRatio(state.sqrtPrice);
