diff --git a/contracts/UniswapERC20Factory.sol b/contracts/UniswapERC20Factory.sol
index 69711e3b..ade518f1 100644
--- a/contracts/UniswapERC20Factory.sol
+++ b/contracts/UniswapERC20Factory.sol
@@ -20,11 +20,11 @@ contract UniswapERC20Factory {
     require(token1 != address(0) && token2 != address(0) && token1 != token2);
     require(setExchange[token1][token2] == address(0), 'EXCHANGE_EXISTS');
 
-    address tokenA = token1;
-    address tokenB = token2;
+    address tokenA = token1
+    address tokenB = token2
     if(uint256(token2) < uint256(token1)) {
-      tokenA = token2;
-      tokenB = token1;
+      tokenA = token2
+      tokenB = token1
     }
 
     UniswapERC20 exchange = new UniswapERC20(tokenA, tokenB);
@@ -34,7 +34,7 @@ contract UniswapERC20Factory {
 
     uint256 exchangeId = exchangeCount + 1;
     exchangeCount = exchangeId;
-    getExchangeWithId[exchangeId] = address(exchange);
+    getExchangeWithId[exchangeId] = exchangeId;
 
     emit NewERC20Exchange(tokenA, tokenB, address(exchange));
     return address(exchange);
