diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index 7e73f01d..2cffc5de 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -49,8 +49,6 @@ contract UniswapV3Pair is IUniswapV3Pair {
     address public immutable override token0;
     address public immutable override token1;
 
-    address public override feeTo;
-
     // ⬇ single storage slot ⬇
     uint112 public override reserve0Virtual;
     uint112 public override reserve1Virtual;
@@ -247,14 +245,6 @@ contract UniswapV3Pair is IUniswapV3Pair {
         }
     }
 
-    function setFeeTo(address feeTo_) external override {
-        require(
-            msg.sender == IUniswapV3Factory(factory).feeToSetter(),
-            'UniswapV3Pair::setFeeTo: caller not feeToSetter'
-        );
-        feeTo = feeTo_;
-    }
-
     // the reason this can't _just_ burn but needs to mint is because otherwise it would incentivize bad starting prices
     function initialize(
         uint112 amount0,
@@ -419,6 +409,7 @@ contract UniswapV3Pair is IUniswapV3Pair {
                 ? FixedPoint.decode144(growthInside.mul(position.liquidityAdjusted)) - position.liquidity
                 : 0;
             if (liquidityFee > 0) {
+                address feeTo = IUniswapV3Factory(factory).feeTo();
                 // take the protocol fee if it's on (feeTo isn't address(0)) and the sender isn't feeTo
                 if (feeTo != address(0) && msg.sender != feeTo) {
                     uint256 liquidityProtocol = liquidityFee / 6;
@@ -679,7 +670,7 @@ contract UniswapV3Pair is IUniswapV3Pair {
         uint112 amount0In,
         address to,
         bytes calldata data
-    ) external override returns (uint112 amount1Out) {
+    ) external returns (uint112 amount1Out) {
         SwapParams memory params = SwapParams({zeroForOne: true, amountIn: amount0In, to: to, data: data});
         return _swap(params);
     }
@@ -689,7 +680,7 @@ contract UniswapV3Pair is IUniswapV3Pair {
         uint112 amount1In,
         address to,
         bytes calldata data
-    ) external override returns (uint112 amount0Out) {
+    ) external returns (uint112 amount0Out) {
         SwapParams memory params = SwapParams({zeroForOne: false, amountIn: amount1In, to: to, data: data});
         return _swap(params);
     }
