diff --git a/contracts/UniswapERC20.sol b/contracts/UniswapERC20.sol
index 5099e305..391695ca 100644
--- a/contracts/UniswapERC20.sol
+++ b/contracts/UniswapERC20.sol
@@ -118,6 +118,8 @@ contract UniswapERC20 is ERC20 {
     uint256 liquidityMinted;
 
     if (_totalSupply > 0) {
+      require(oldReserveA > 0, "INVALID_TOKEN_A_RESERVE");
+      require(oldReserveB > 0, "INVALID_TOKEN_B_RESERVE");
       liquidityMinted = Math.min((amountA.mul(_totalSupply).div(oldReserveA)), (amountB.mul(_totalSupply).div(oldReserveB)));
     } else {
       liquidityMinted = Math.sqrt(amountA.mul(amountB));
