diff --git a/contracts/UniswapERC20.sol b/contracts/UniswapERC20.sol
index 7d9335e9..016f836b 100644
--- a/contracts/UniswapERC20.sol
+++ b/contracts/UniswapERC20.sol
@@ -50,19 +50,19 @@ contract UniswapERC20 is ERC20 {
 
   //TO: DO msg.sender is wrapper
   function swapAForB(uint256 amountSold, address recipient) public nonReentrant returns (uint256) {
-      uint256 amountBought = swap(tokenA, tokenB, amountSold, recipient);
+      amountBought = swap(tokenA, tokenB, amountSold);
       emit SwapAForB(msg.sender, amountSold, amountBought);
       return amountBought;
   }
 
   //TO: DO msg.sender is wrapper
   function swapBForA(uint256 amountSold, address recipient) public nonReentrant returns (uint256) {
-      uint256 amountBought = swap(tokenB, tokenA, amountSold, recipient);
+      amountBought = swap(tokenB, tokenA, amountSold);
       emit SwapBForA(msg.sender, amountSold, amountBought);
       return amountBought;
   }
 
-  function swap(address inputToken, address outputToken, uint256 amountSold, address recipient) private returns (uint256) {
+  function swap(address inputToken, address outputToken, uint256 amountSold) private returns (uint256) {
       uint256 inputReserve = IERC20(inputToken).balanceOf(address(this));
       uint256 outputReserve = IERC20(outputToken).balanceOf(address(this));
       uint256 amountBought = getInputPrice(amountSold, inputReserve, outputReserve);
