diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index dcf358d1..9f0932cf 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -217,26 +217,28 @@ contract UniswapV3Pair is IUniswapV3Pair {
         if (liquidityDelta != 0) {
             uint32 blockTimestamp = _blockTimestamp();
 
-            flippedLower = ticks.update(
-                tickLower,
-                tick,
-                liquidityDelta,
-                _feeGrowthGlobal0X128,
-                _feeGrowthGlobal1X128,
-                blockTimestamp,
-                false,
-                maxLiquidityPerTick
-            );
-            flippedUpper = ticks.update(
-                tickUpper,
-                tick,
-                liquidityDelta,
-                _feeGrowthGlobal0X128,
-                _feeGrowthGlobal1X128,
-                blockTimestamp,
-                true,
-                maxLiquidityPerTick
-            );
+            flippedLower =
+                ticks.update(
+                    tickLower,
+                    tick,
+                    liquidityDelta,
+                    _feeGrowthGlobal0X128,
+                    _feeGrowthGlobal1X128,
+                    blockTimestamp,
+                    false,
+                    maxLiquidityPerTick
+                );
+            flippedUpper =
+                ticks.update(
+                    tickUpper,
+                    tick,
+                    liquidityDelta,
+                    _feeGrowthGlobal0X128,
+                    _feeGrowthGlobal1X128,
+                    blockTimestamp,
+                    true,
+                    maxLiquidityPerTick
+                );
 
             if (flippedLower) tickBitmap.flipTick(tickLower, tickSpacing);
             if (flippedUpper) tickBitmap.flipTick(tickUpper, tickSpacing);
