diff --git a/contracts/UniswapV2Factory.sol b/contracts/UniswapV2Factory.sol
index a66ad824..78607164 100644
--- a/contracts/UniswapV2Factory.sol
+++ b/contracts/UniswapV2Factory.sol
@@ -2,6 +2,7 @@ pragma solidity =0.5.16;
 
 import './interfaces/IUniswapV2Factory.sol';
 import './UniswapV2Pair.sol';
+import './libraries/PairNamer.sol';
 
 contract UniswapV2Factory is IUniswapV2Factory {
     address public feeTo;
@@ -10,8 +11,6 @@ contract UniswapV2Factory is IUniswapV2Factory {
     mapping(address => mapping(address => address)) public getPair;
     address[] public allPairs;
 
-    event PairCreated(address indexed token0, address indexed token1, address pair, uint);
-
     constructor(address _feeToSetter) public {
         feeToSetter = _feeToSetter;
     }
@@ -37,6 +36,17 @@ contract UniswapV2Factory is IUniswapV2Factory {
         emit PairCreated(token0, token1, pair, allPairs.length);
     }
 
+    function setPairNameAndSymbol(address tokenA, address tokenB) external {
+        (address token0, address token1) = tokenA < tokenB ? (tokenA, tokenB) : (tokenB, tokenA);
+        address pair = getPair[token0][token1];
+        require(pair != address(0), 'UniswapV2: PAIR_NOT_EXISTS'); // single check is sufficient
+
+        IUniswapV2Pair(pair).initializeNameAndSymbol(
+            PairNamer.pairName(token0, token1),
+            PairNamer.pairSymbol(token0, token1)
+        );
+    }
+
     function setFeeTo(address _feeTo) external {
         require(msg.sender == feeToSetter, 'UniswapV2: FORBIDDEN');
         feeTo = _feeTo;
