diff --git a/contracts/test/UniswapV3PairEchidnaTest.sol b/contracts/test/UniswapV3PairEchidnaTest.sol
index bce3a2fb..18625ba2 100644
--- a/contracts/test/UniswapV3PairEchidnaTest.sol
+++ b/contracts/test/UniswapV3PairEchidnaTest.sol
@@ -46,12 +46,18 @@ contract UniswapV3PairEchidnaTest {
         pair.initialize(sqrtPrice, '');
     }
 
-    function swapExactInput(bool zeroForOne, uint256 amountIn) external {
-        require(amountIn > 0);
-        require(amountIn < 1e18);
-        if (zeroForOne) token0.transfer(address(payer), amountIn);
-        else token1.transfer(address(payer), amountIn);
-        pair.swap(zeroForOne, int256(amountIn), address(this), '');
+    function swapExact0For1(uint256 amount0In) external {
+        require(amount0In > 1);
+        require(amount0In < 1e18);
+        token0.transfer(address(payer), amount0In);
+        pair.swapExact0For1(amount0In, address(this), '');
+    }
+
+    function swapExact1For0(uint256 amount1In) external {
+        require(amount1In > 1);
+        require(amount1In < 1e18);
+        token1.transfer(address(payer), amount1In);
+        pair.swapExact1For0(amount1In, address(this), '');
     }
 
     function mint(
