diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index 3509f43a..67e03e2d 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -21,7 +21,6 @@ import './libraries/PriceMath.sol';
 import './interfaces/IUniswapV3Pair.sol';
 import './interfaces/IUniswapV3Factory.sol';
 import './interfaces/IUniswapV3Callee.sol';
-import './libraries/TickBitMap.sol';
 
 contract UniswapV3Pair is IUniswapV3Pair {
     using SafeMath for uint112;
@@ -33,7 +32,6 @@ contract UniswapV3Pair is IUniswapV3Pair {
     using SafeCast for uint256;
     using MixedSafeMath for uint112;
     using FixedPoint for FixedPoint.uq112x112;
-    using TickBitMap for uint256[58];
 
     // Number of fee options
     uint8 public constant override NUM_FEE_OPTIONS = 6;
@@ -61,9 +59,6 @@ contract UniswapV3Pair is IUniswapV3Pair {
     // TODO figure out the best way to pack state variables
     address public override feeTo;
 
-    // see TickBitMap.sol
-    uint256[58] public override tickBitMap;
-
     // meant to be accessed via getPriceCumulative
     FixedPoint.uq144x112 private price0CumulativeLast; // cumulative (token1 / token0) oracle price
     FixedPoint.uq144x112 private price1CumulativeLast; // cumulative (token0 / token1) oracle price
@@ -376,12 +371,6 @@ contract UniswapV3Pair is IUniswapV3Pair {
             tickInfo.secondsOutside = _blockTimestamp();
         }
         tickInfo.numPositions = 1;
-        tickBitMap.flipTick(tick);
-    }
-
-    function _clearTick(int16 tick) private {
-        delete tickInfos[tick];
-        tickBitMap.flipTick(tick);
     }
 
     function initialize(int16 tick, uint8 feeVote) external override lock {
@@ -521,9 +510,9 @@ contract UniswapV3Pair is IUniswapV3Pair {
 
             // if necessary, uninitialize both ticks and increment the position counter
             if (position.liquidity == 0 && params.liquidityDelta < 0) {
-                if (tickInfoLower.numPositions == 1) _clearTick(params.tickLower);
+                if (tickInfoLower.numPositions == 1) delete tickInfos[params.tickLower];
                 else tickInfoLower.numPositions--;
-                if (tickInfoUpper.numPositions == 1) _clearTick(params.tickUpper);
+                if (tickInfoUpper.numPositions == 1) delete tickInfos[params.tickUpper];
                 else tickInfoUpper.numPositions--;
 
                 // reset fee growth
