diff --git a/contracts/test/UniswapV3PairEchidnaTest.sol b/contracts/test/UniswapV3PairEchidnaTest.sol
index c1be57d0..1a6a11df 100644
--- a/contracts/test/UniswapV3PairEchidnaTest.sol
+++ b/contracts/test/UniswapV3PairEchidnaTest.sol
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier: UNLICENSED
+// SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity =0.7.6;
 
 import '@uniswap/lib/contracts/libraries/FullMath.sol';
