diff --git a/contracts/UniswapETH.sol b/contracts/UniswapETH.sol
index 8695931a..e087007f 100644
--- a/contracts/UniswapETH.sol
+++ b/contracts/UniswapETH.sol
@@ -1,8 +1,8 @@
 pragma solidity ^0.5.11;
 import './ERC20.sol';
 import './interfaces/IERC20.sol';
-import './interfaces/IUniswapETHFactory.sol';
-import './interfaces/IUniswapETH.sol';
+import './interfaces/IUniswapFactory.sol';
+import './interfaces/IUniswapExchange.sol';
 
 
 contract UniswapETH is ERC20 {
