diff --git a/contracts/UniswapV3Pair.sol b/contracts/UniswapV3Pair.sol
index 816be27b..5c55398b 100644
--- a/contracts/UniswapV3Pair.sol
+++ b/contracts/UniswapV3Pair.sol
@@ -254,7 +254,7 @@ contract UniswapV3Pair is IUniswapV3Pair {
         Aggregate memory deltaFeeVote;
         require(feeVote > MIN_FEEVOTE && feeVote < MAX_FEEVOTE, "UniswapV3: INVALID_FEE_VOTE");
         require(lowerTick < upperTick, "UniswapV3: BAD_TICKS");
-        { // scope to help with compilation
+        { // scope to allow compilation
         bytes32 positionKey = keccak256(abi.encodePacked(msg.sender, lowerTick, upperTick));
         require(virtualSupply > 0, 'UniswapV3: NOT_INITIALIZED');
         Position memory _position = positions[positionKey];
