diff --git a/contracts/UniswapV2Factory.sol b/contracts/UniswapV2Factory.sol
index a66ad824..99d6c0a6 100644
--- a/contracts/UniswapV2Factory.sol
+++ b/contracts/UniswapV2Factory.sol
@@ -10,8 +10,6 @@ contract UniswapV2Factory is IUniswapV2Factory {
     mapping(address => mapping(address => address)) public getPair;
     address[] public allPairs;
 
-    event PairCreated(address indexed token0, address indexed token1, address pair, uint);
-
     constructor(address _feeToSetter) public {
         feeToSetter = _feeToSetter;
     }
