diff --git a/contracts/UniswapV2Exchange.sol b/contracts/UniswapV2Exchange.sol
index 7c8bb766..4b95c260 100644
--- a/contracts/UniswapV2Exchange.sol
+++ b/contracts/UniswapV2Exchange.sol
@@ -87,7 +87,7 @@ contract UniswapV2Exchange is IUniswapV2Exchange, UniswapV2ERC20 {
         emit Sync(reserve0, reserve1);
     }
 
-    // if fee is on, mint liquidity equivalent to 1/6th of the growth in sqrt(k)
+    // if fee is on, mint liquidity equivalent to 20% of growth in sqrt(k)
     function _mintFee(uint112 _reserve0, uint112 _reserve1) private returns (bool feeOn) {
         address feeTo = IUniswapV2Factory(factory).feeTo();
         feeOn = feeTo != address(0);
