diff --git a/contracts/test/UnsafeMathEchidnaTest.sol b/contracts/test/UnsafeMathEchidnaTest.sol
index edc65e3b..08f2a154 100644
--- a/contracts/test/UnsafeMathEchidnaTest.sol
+++ b/contracts/test/UnsafeMathEchidnaTest.sol
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity =0.8.19;
 
-import {UnsafeMath} from "../libraries/UnsafeMath.sol";
+import {UnsafeMath} from '../libraries/UnsafeMath.sol';
 
 contract UnsafeMathEchidnaTest {
     function checkDivRoundingUp(uint256 x, uint256 d) external pure {
